<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Battenberg subclonal copy number caller • Battenberg</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Battenberg subclonal copy number caller">
<meta property="og:description" content="Estimate subclonal copy number from whole genome sequencing or SNP6 data.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="battenberg">Battenberg<a class="anchor" aria-label="anchor" href="#battenberg"></a>
</h1></div>
<p>This repository contains code for the whole genome sequencing subclonal copy number caller Battenberg, as described in <a href="https://www.ncbi.nlm.nih.gov/pubmed/22608083" class="external-link">Nik-Zainal, Van Loo, Wedge, et al. (2012), Cell</a>.</p>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>The instructions below will install the latest stable Battenberg version.</p>
<div class="section level4">
<h4 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h4>
<p>Installing from Github requires devtools and Battenberg requires the modified copynumber package from “igordot/copynumber” and readr, gtools, splines, ggplot2, gridExtra, RColorBrewer, VariantAnnotation, GenomicRanges and ASCAT. The pipeline requires parallel and doParallel. From the command line run:</p>
<pre><code>R -q -e 'BiocManager::install(c("devtools", "splines", "readr", "doParallel", "ggplot2", "RColorBrewer", "gridExtra", "gtools", "parallel", "igordot/copynumber", "VariantAnnotation", "GenomicRanges"))'
R -q -e 'devtools::install_github("VanLoo-lab/ascat/ASCAT")'</code></pre>
</div>
<div class="section level4">
<h4 id="installation-from-github">Installation from Github<a class="anchor" aria-label="anchor" href="#installation-from-github"></a>
</h4>
<p>To install Battenberg, run the following from the command line:</p>
<pre><code>R -q -e 'devtools::install_github("Wedge-Oxford/battenberg")'</code></pre>
</div>
<div class="section level4">
<h4 id="required-reference-files">Required reference files<a class="anchor" aria-label="anchor" href="#required-reference-files"></a>
</h4>
<p><code>GRCh37</code> reference files may downloaded from here: <a href="https://ora.ox.ac.uk/objects/uuid:2c1fec09-a504-49ab-9ce9-3f17bac531bc" class="external-link uri">https://ora.ox.ac.uk/objects/uuid:2c1fec09-a504-49ab-9ce9-3f17bac531bc</a></p>
<p>The bundle contains the following files: * battenberg_1000genomesloci2012_v3.tar.gz * battenberg_impute_1000G_v3.tar.gz * probloci_270415.txt.gz * battenberg_wgs_gc_correction_1000g_v3.tar.gz * battenberg_wgs_replic_correction_1000g_v3.tar.gz * battenberg_snp6_exe.tgz (SNP6 only) * battenberg_snp6_ref.tgz (SNP6 only)</p>
<p><code>GRCh38</code> reference files may be downloaded from here: <a href="https://ora.ox.ac.uk/objects/uuid:08e24957-7e76-438a-bd38-66c48008cf52" class="external-link uri">https://ora.ox.ac.uk/objects/uuid:08e24957-7e76-438a-bd38-66c48008cf52</a></p>
<p>The bundle contains the following files: * 1000G_loci_hg38.zip * imputation.zip * beagle5.zip * probloci.zip * GC_correction_hg38.zip * RT_correction_hg38.zip * README.txt</p>
</div>
<div class="section level4">
<h4 id="pipeline">Pipeline<a class="anchor" aria-label="anchor" href="#pipeline"></a>
</h4>
<p>Go into <code>inst/example</code> for example WGS and SNP6 R-only pipelines.</p>
</div>
</div>
<div class="section level2">
<h2 id="description-of-the-output">Description of the output<a class="anchor" aria-label="anchor" href="#description-of-the-output"></a>
</h2>
<div class="section level3">
<h3 id="key-output-files">Key output files<a class="anchor" aria-label="anchor" href="#key-output-files"></a>
</h3>
<ul>
<li>
<code>[samplename]_copynumber.txt</code> contains the copy number data (see table below)</li>
<li>
<code>[samplename]_rho_and_psi.txt</code> contains the purity estimate (make sure to use the FRAC_genome, rho field in the second row, first column)</li>
<li>
<code>[samplename]_BattenbergProfile*png</code> shows the profile (the two variants show subclonal copy number in a different way)</li>
<li>
<code>[samplename]_subclones_chr*.png</code> show detailed figures of the copy number calls per chromosome</li>
<li>
<code>[samplename]_distance.png</code> This shows the purity and ploidy solution space and can be used to pick alternative solutions</li>
</ul>
<p>The copy number profile saved in the <code>[samplename]_copynumber.txt</code> is a tab delimited file in text format. Within this file there is a line for each segment in the tumour genome. Each segment will have either one or two copy number states:</p>
<ul>
<li>If there is one state that line represents the clonal copy number (i.e. all tumour cells have this state)</li>
<li>If there are two states that line represents subclonal copy number (i.e. there are two populations of cells, each with a different state)</li>
</ul>
<p>A copy number state consists of a major and a minor allele and their frequencies, which together add give the total copy number for that segment and an estimate fraction of tumour cells that carry each allele.</p>
<p>The following columns are available in the Battenberg output:</p>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Column</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>chr</td>
<td>The chromosome of the segment</td>
</tr>
<tr class="even">
<td>startpos</td>
<td>Start position on the chromosome</td>
</tr>
<tr class="odd">
<td>endpos</td>
<td>End position on the chromosome</td>
</tr>
<tr class="even">
<td>BAF</td>
<td>The B-allele frequency of the segment</td>
</tr>
<tr class="odd">
<td>pval</td>
<td>P-value that is obtained when testing whether this segment should be represented by one or two states. A low p-value will result in the fitting of a second copy number state</td>
</tr>
<tr class="even">
<td>LogR</td>
<td>The log ratio of normalised tumour coverage versus its matched normal sequencing sample</td>
</tr>
<tr class="odd">
<td>ntot</td>
<td>An internal total copy number value used to determine the priority of solutions. NOTE: This is not the total copy number of this segment!</td>
</tr>
<tr class="even">
<td>nMaj1_A</td>
<td>The major allele copy number of state 1 from solution A</td>
</tr>
<tr class="odd">
<td>nMin1_A</td>
<td>The minor allele copy number of state 1 from solution A</td>
</tr>
<tr class="even">
<td>frac1_A</td>
<td>Fraction of tumour cells carrying state 1 in solution A</td>
</tr>
<tr class="odd">
<td>nMaj2_A</td>
<td>The major allele copy number of state 2 from solution A. This value can be NA</td>
</tr>
<tr class="even">
<td>nMin2_A</td>
<td>The minor allele copy number of state 2 from solution A. This value can be NA</td>
</tr>
<tr class="odd">
<td>frac2_A</td>
<td>Fraction of tumour cells carrying state 2 in solution A. This value can be NA</td>
</tr>
<tr class="even">
<td>SDfrac_A</td>
<td>Standard deviation on the BAF of SNPs in this segment, can be used as a measure of uncertainty</td>
</tr>
<tr class="odd">
<td>SDfrac_A_BS</td>
<td>Bootstrapped standard deviation</td>
</tr>
<tr class="even">
<td>frac1_A_0.025</td>
<td>Associated 95% confidence interval of the bootstrap measure of uncertainty</td>
</tr>
</tbody>
</table>
<p>Followed by possible equivalent solutions B to F with the same columns as defined above for solution A (due to the way a profile is fit Battenberg can generate a series of equivalent solutions that are reported separately in the output).</p>
</div>
<div class="section level3">
<h3 id="plots-for-qc">Plots for QC<a class="anchor" aria-label="anchor" href="#plots-for-qc"></a>
</h3>
<p>It also produces a number plots that show the raw data and are useful for QC (and their raw data files denoted by *.tab)</p>
<ul>
<li>
<code>[samplename].tumour.png</code> and <code>[samplename].germline.png</code> show the raw BAF and logR</li>
<li>
<code>[samplename]_coverage.png</code> contains coverage divided by the mean coverage of both tumour and normal</li>
<li>
<code>[samplename]_alleleratio.png</code> shows BAF*logR, a rough approximation of what the data looks like shortly before copy number calling</li>
</ul>
</div>
<div class="section level3">
<h3 id="intermediate-figures">Intermediate figures<a class="anchor" aria-label="anchor" href="#intermediate-figures"></a>
</h3>
<p>Finally, a range of plots show intermediate steps and can occasionally be useful</p>
<ul>
<li>
<code>[samplename]_chr*_heterozygousData.png</code> shows reconstructed haplotype blocks in the characteristic Battenberg cake pattern</li>
<li>
<code>[samplename]_RAFseg_chr*.png</code> and <code>[samplename]_segment_chr*.png</code> contains segmentation data for step 1 and step 2 respectively</li>
<li>
<code>[samplename]_nonroundedprofile.png</code> shows the copy number profile without rounding to integers</li>
<li>
<code>[samplename]_copynumberprofile.png</code> shows the copy number profile with (including subclonal copy number) rounding to integers</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="advice-for-including-structural-variant-breakpoints">Advice for including structural variant breakpoints<a class="anchor" aria-label="anchor" href="#advice-for-including-structural-variant-breakpoints"></a>
</h2>
<p>Battenberg can take prior breakpoints, from structural variants (SVs) for example, as input. SV breakpoints are typically much more precise and a pair of SVs can be closer together then what typically can be obtained from a BAF or coverage track. It is therefore adventageous to include prior breakpoints in a Battenberg run. However, including too many (as in 100s) incorrect breakpoints can have adverse effects by allowing many small segments to be affected by noise where there isn’t any signal and increasing the runtime of the pipeline. It is therefore advised to <code>filter prior breakpoints from SVs such that the genome is slightly oversegmented.</code> Finally, some SV types, such as inversions, do not constitute a change in copy number and therefore also add breakpoints that should not be considered. It is therefore also advised to <code>filter breakpoints from SVs that do not cause a change in copynumber, such as inversions</code>. Please note that the chromosome names in the SV file <strong>do not</strong> include the “chr” prefix.</p>
</div>
<div class="section level2">
<h2 id="building-a-release">Building a release<a class="anchor" aria-label="anchor" href="#building-a-release"></a>
</h2>
<p>In RStudio: In the Build tab, click Check Package</p>
<p>Then open the <code>NAMESPACE</code> file and edit:</p>
<pre><code><span><span class="fu">S3method</span><span class="op">(</span><span class="va">plot</span>,<span class="va">haplotype.data</span><span class="op">)</span></span></code></pre>
<p>to:</p>
<pre><code><span><span class="fu">export</span><span class="op">(</span><span class="va">plot.haplotype.data</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="hg38-for-beagle5">hg38 for Beagle5<a class="anchor" aria-label="anchor" href="#hg38-for-beagle5"></a>
</h2>
<p>Modified original code to derive the input vcf for Beagle5 and hg38:</p>
<pre><code>#!/bin/bash
#
# READ_ME file (08 Dec 2015)
#
# 1000 Genomes Project Phase 3 data release (version 5a) in VCF format for use with Beagle version 4.x
#    Data Source: ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/*
#
# NOTES:
#
# 1) Markers with &lt;5 copies of the reference allele or &lt;5 copies of the non-reference alleles have been excluded.
#
# 2) Structural variants have been excluded.
#
# 3) All non-unique identifiers in the ID column are removed from the ID column
#
# 4) Additional marker filtering may be performed using the gtstats.jar and filterlines.jar utilities
#
# 5) Sample information is in files:
#      integrated_call_samples.20130502.ALL.ped
#      integrated_call_samples_v3.20130502.ALL.panel
#
# 6) Male haploid chromosome X genotypes are encoded as diploid homozygous genotypes.
#
############################################################################
#  The following shell script was used to create the files in this folder  #
############################################################################
#

## required if loading modules
module load Java
module load HTSlib

## wget for GRCh37
## wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/*

# wget for GRCh38 (liftover from hg38)
# wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/supporting/GRCh38_positions/*
# see article: https://wellcomeopenresearch.org/articles/4-50
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000_genomes_project/release/20190312_biallelic_SNV_and_INDEL/ALL.chr*
wget http://bochet.gcc.biostat.washington.edu/beagle/1000_Genomes_phase3_v5a/utilities/filterlines.jar
wget http://bochet.gcc.biostat.washington.edu/beagle/1000_Genomes_phase3_v5a/utilities/gtstats.jar
wget http://bochet.gcc.biostat.washington.edu/beagle/1000_Genomes_phase3_v5a/utilities/remove.ids.jar
wget http://bochet.gcc.biostat.washington.edu/beagle/1000_Genomes_phase3_v5a/utilities/simplify-vcf.jar

## Downloadable from Beagle5 web page
gts="java -ea -jar gtstats.jar"
fl="java -ea -jar filterlines.jar"
rmids="java -ea -jar remove.ids.jar"
simplify="java -ea -jar simplify-vcf.jar"
min_minor="5"

## Running directory
src="./"
#mkdir ${src}
#cd ${src}
#cd -

## Go through autosomes and prepare vcf
for chr in $(seq 4 5); do
echo "chr${chr}"
#input="${src}ALL.chr${chr}_GRCh38.genotypes.20170504.vcf.gz"
input="${src}ALL.chr${chr}.shapeit2_integrated_snvindels_v2a_27022019.GRCh38.phased.vcf.gz"
#vcf_removefield="${input}_removedfield.vcg.gz"
vcf="chr${chr}.1kg.phase3.v5a_GRCh38.vcf.gz"
excl="chr${chr}.1kg.phase3.v5a_GRCh38.excl"

#zcat ${input} | awk '/^[^#]/ {gsub(";GRC.*","",$9);print}' &gt; ${vcf_removefield}
zcat ${input} | grep -v 'SVTYPE' | ${gts} | ${fl} \# -13 ${min_minor} 1000000 | cut -f2 &gt; ${excl}
zcat ${input} | grep -v 'SVTYPE' | grep -v '^#' | cut -f3 | tr ';' '\n' | sort | uniq -d &gt; chr${chr}.dup.id

# BEGIN: add 4 duplicate markers to exclusion list
#if [ ${chr} == "8" ]; then echo "88316919"; fi &gt;&gt; ${excl}
#if [ ${chr} == "12" ]; then echo ""; fi &gt;&gt; ${excl}
#if [ ${chr} == "14" ]; then echo "21181798"; fi  &gt;&gt; ${excl}
#if [ ${chr} == "17" ]; then echo "1241338"; fi  &gt;&gt; ${excl}
# END:  add 4 duplicate markers to exclusion list

zcat ${input} | grep -v 'SVTYPE' | ${fl} \# \-2 ${excl} | ${simplify} | ${rmids} chr${chr}.dup.id | bgzip -c &gt; ${vcf}
tabix ${vcf}
done

## Same for chromosome X
chr="X"
#in="${src}ALL.chr${chr}_GRCh38.genotypes.20170504.vcf.gz"
in="${src}ALL.chr${chr}.shapeit2_integrated_snvindels_v2a_27022019.GRCh38.phased.vcf.gz"
vcf="chr${chr}.1kg.phase3.v5a_GRCh38.vcf.gz"
excl="chr${chr}.1kg.phase3.v5a_GRCh38.excl"

### sed command recodes male chrom X genotypes as homozygote diploid genotypes
### sed command makes temporary change to floating point numbers and diploid genotypes to permit use of word-boundary '\&gt;'
cat_in=" zcat ${in} | grep -v 'SVTYPE' | sed \
-e 's/\t0\t\.\t/\t111\tPASS\t/' \
-e 's/0\tPASS/222\tPASS/' \
-e 's/\([0-9]\)\./\1WXYZ/g' \
-e 's/\([0-9]\)|\([0-9]\)/\1X\2/g' \
-e 's/\t\([0-9]\)\&gt;/\t\1|\1/g' \
-e 's/\([0-9]\)WXYZ/\1./g' \
-e 's/\([0-9]\)X\([0-9]\)/\1|\2/g';"

echo ${chr}
  eval ${cat_in} | grep -v '^#' | cut -f3 | tr ';' '\n' | sort | uniq -d &gt; chr${chr}.dup.id
  eval ${cat_in} | ${gts} | ${fl} '\#' -13 ${min_minor} 1000000 | cut -f2 &gt; ${excl}

  # BEGIN: add duplicate markers to exclusion list
  #if [ ${chr} == "X" ]; then echo "5457254"; fi &gt;&gt; ${excl}
  #if [ ${chr} == "X" ]; then echo "32344545"; fi &gt;&gt; ${excl}
  #if [ ${chr} == "X" ]; then echo "68984437"; fi &gt;&gt; ${excl}
  # END:  add duplicate markers to exclusion list

eval ${cat_in} | ${fl} \# \-2 ${excl} | ${simplify} | ${rmids} chr${chr}.dup.id | bgzip -c &gt; ${vcf}
tabix ${vcf}</code></pre>
<p>Run R code to generate loci, allele and gc_content files:</p>
<pre><code>##########################################################################
## set working directory to where the vcf files are located
setwd("./")
##########################################################################
ffs &lt;- dir(full=T,pattern="1kg.phase3.v5a_GRCh38.vcf.gz$")
MC.CORES &lt;- 10 ## set number of cores to use
##########################################################################


##########################################################################
library(parallel)
##########################################################################
## Further remove unreferenced alleles and duplicates
##########################################################################
mclapply(ffs[length(ffs):1],function(x)
{
    cat(paste(x,"read file"))
    out  &lt;- gsub(".vcf.gz","nounref.vcf.gz",x)
    cmd &lt;- paste0("zcat ",
                  x,
                  " | grep -v '",
                  "\\",
                  ".",
                  "|","' ",
                  " | grep -v '",
                  "|",
                  "\\",
                  ".",
                  "' "
                 ,"|"," awk '!seen[$2]++' |  gzip &gt; ",
                  out)
    system(cmd)
},mc.cores=MC.CORES)
##########################################################################

ffs &lt;- dir(full=T,pattern="1kg.phase3.v5a_GRCh38nounref.vcf.gz$")
## Generate Loci files
##########################################################################
mclapply(ffs[length(ffs):1],function(x)
{
    cat(paste(x,"read file"))
    out  &lt;- gsub(".vcf.gz","_loci.txt",x)
    cmd &lt;- paste0("zcat ",
                  x," | grep -v '^#' | awk -v OFS='\t' '{print $1, $2}' &gt; ",
                  out)
    system(cmd)
},mc.cores=MC.CORES)
##########################################################################
## with chr string (for BAMs that contain "chr")
mclapply(ffs[length(ffs):1],function(x)
{
    ##cat(paste(x,"read file"))
    out  &lt;- gsub(".vcf.gz","_loci_chrstring.txt",x)
    cmd &lt;- paste0("zcat ",
                  x," | grep -v '^#' | awk -v OFS='\t' '{print ",
                  '"chr"'
                 ,"$1, $2}' &gt; ",
                  out)
    system(noquote(cmd))
},mc.cores=MC.CORES)
##########################################################################
## Generate Allele Files
##########################################################################
mclapply(ffs[length(ffs):1],function(x)
{
    cat(paste(x,"read file"))
    out  &lt;- gsub(".vcf.gz","_allele_letter.txt",x)
    cmd &lt;- paste0("zcat ",
                  x," | grep -v '^#' | awk -v OFS='\t' '{print $2, $4, $5}' &gt; ",
                  out)
    system(cmd)
},mc.cores=MC.CORES)
##########################################################################
## Convert Alleles into Indices for BB input
indices &lt;- c("A"=1,"C"=2,"G"=3,"T"=4)
##########################################################################
mclapply(ffs[length(ffs):1],function(x)
{
    cat(".")
    inp &lt;- gsub(".vcf.gz","_allele_letter.txt",x)
    out &lt;- gsub(".vcf.gz","_allele_index.txt",x)
    df &lt;- as.data.frame(data.table::fread(inp))
    ref &lt;- indices[df[,2]]
    alt &lt;- indices[df[,3]]
    ndf &lt;- data.frame(position=df[,1],
                      a0=ref,
                      a1=alt)
    write.table(ndf,file=out,
                row.names=F,col.names=T,sep="\t",quote=F)
},mc.cores=5)
##########################################################################


##########################################################################
## Symlink loci to change the names allowing for a "prefix" before chr in BB
##########################################################################
allfs &lt;- dir(full=T)
allfs_loci &lt;- allfs[grepl("loci.txt$",allfs)]
tnull &lt;- lapply(allfs_loci,function(x)
{
    cmd &lt;- paste0("ln -s ",x," ",gsub("chr(.*?)\\.(.*).txt","\\2_chr\\1.txt",x))
    system(cmd)
    if(grepl("chrX",x))
    {
        cmd &lt;- paste0("ln -s ",x," ",gsub("chr(.*?)\\.(.*).txt","\\2_chr23.txt",x))
        system(cmd)
    }
})
##########################################################################
allfs &lt;- dir(full=T)
allfs_loci &lt;- allfs[grepl("loci_chrstring.txt$",allfs)]
tnull &lt;- lapply(allfs_loci,function(x)
{
    cmd &lt;- paste0("ln -s ",x," ",gsub("chr(.*?)\\.(.*).txt","\\2_chr\\1.txt",x))
    system(cmd)
    if(grepl("chrX",x))
    {
        cmd &lt;- paste0("ln -s ",x," ",gsub("chr(.*?)\\.(.*).txt","\\2_chr23.txt",x))
        system(cmd)
    }
})
##########################################################################
## Symlink alleles: same as for loci, symlink to change name for the
## use of prefixes
##########################################################################
allfs &lt;- dir(full=T)
allfs_index &lt;- allfs[grepl("allele_index",allfs)]
tnull &lt;- lapply(allfs_index,function(x)
{
    cmd &lt;- paste0("ln -s ",x," ",gsub("chr(.*?)\\.(.*).txt","\\2_chr\\1.txt",x))
    system(cmd)
    if(grepl("chrX",x))
    {
        cmd &lt;- paste0("ln -s ",x," ",gsub("chr(.*?)\\.(.*).txt","\\2_chr23.txt",x))
        system(cmd)
    }
})

##########################################################################
##  Derive GC content files
##########################################################################
library(Rsamtools)
library(data.table)
library(Biostrings)
##########################################################################
gcTrack &lt;- function(chr,
                    pos,
                    dna,
                    window=5000)
{
    gc &lt;- rowSums(letterFrequencyInSlidingView(dna[[chr]],
                                               window,
                                               c("G","C")))/window
    gc[pos]
}
getRefGenome &lt;- function (fasta = FASTA, CHRS = paste0("", c(1:22, "X", "Y",
    "MT")))
{
    dna &lt;- Biostrings::readDNAStringSet(fasta, format = "fasta")
    dna &lt;- lapply(1:length(CHRS), function(x) dna[[x]])
    names(dna) &lt;- CHRS
    return(dna)
}
##########################################################################
FASTA &lt;- "genome.fa" ## Link to genome reference fasta file
CHRS &lt;- paste0("", c(1:22, "X"))
BEAGLELOCI.template &lt;- "chrCHROMNAME.1kg.phase3.v5a_GRCh38nounref_loci.txt"
##########################################################################
REFGENOME &lt;- getRefGenome(fasta = FASTA, CHRS = CHRS) ## Loads genome
in memory
##########################################################################
OUTDIR &lt;- "1000genomes_2012_v3_gcContent_hg38"
system(paste0("mkdir ",OUTDIR))
##########################################################################

##########################################################################
windows &lt;- c(25,
             50,
             100,
             200,
             500,
             1000,
             2000,
             5000,
             10000,
             20000,
             50000,
             100000,
             200000,
             500000,
             1000000,
             2000000,
             5000000,
             10000000)
names(windows) &lt;- formatC(windows,format="f",digits=0)
names(windows) &lt;- gsub("000000$","Mb",names(windows))
names(windows) &lt;- gsub("000$","kb",names(windows))
names(windows) &lt;- sapply(names(windows),function(x) if(grepl("[0-9]$",x)) paste0(x,"bp") else x)
##########################################################################

writeGC &lt;- function(gccontent,chr,outdir)
{
    write.table(gccontent,
                file=gzfile(paste0(outdir,"/1000_genomes_GC_corr_chr_",chr,".txt.gz")),
                col.names=T,
                row.names=T,quote=F,sep="\t")
}

mclapply(CHRS,function(chr)
{
    cat(chr)
    snppos &lt;- as.data.frame(data.table::fread(gsub("CHROMNAME",chr,BEAGLELOCI.template)))
    gccontent &lt;- sapply(windows,function(window) gcTrack(chr=chr,
                                                         pos=snppos[,2],
                                                         dna=REFGENOME,
                                                         window=window))*100
    gccontent &lt;- cbind(rep(chr,nrow(gccontent)),snppos[,2],gccontent)
    colnames(gccontent)[1:2] &lt;- c("chr","Position")
    rownames(gccontent) &lt;- snppos[,2]
    writeGC(gccontent,chr,OUTDIR)
},mc.cores=10)
   </code></pre>
<div class="section level4">
<h4 id="example-run">Example run<a class="anchor" aria-label="anchor" href="#example-run"></a>
</h4>
<p>To run using Beagle5, simply parametrise the same way you would run under impute2. It should be back compatible, so you can run impute2 by setting usebeagle=FALSE. And it uses the same input files needed for the pipeline, i.e. 1000G loci/alleles + ref panel + prob loci + imputeinfo file etc.</p>
<p>The map plink files for Beagle can be downloaded from: <a href="http://bochet.gcc.biostat.washington.edu/beagle/genetic_maps/" class="external-link uri">http://bochet.gcc.biostat.washington.edu/beagle/genetic_maps/</a></p>
<pre><code><span><span class="va">BEAGLEJAR</span> <span class="op">&lt;-</span> <span class="st">"$PATHTOBEAGLEFILES/beagle.24Aug19.3e8.jar"</span></span>
<span><span class="va">BEAGLEREF.template</span> <span class="op">&lt;-</span> <span class="st">"$PATHTOBEAGLEFILES/chrCHROMNAME.1kg.phase3.v5a.b37.bref3"</span></span>
<span><span class="va">BEAGLEPLINK.template</span> <span class="op">&lt;-</span> <span class="st">"$PATHTOBEAGLEFILES/plink.chrCHROMNAME.GRCh37.map"</span></span>
<span></span>
<span><span class="va">timed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="reference/battenberg.html">battenberg</a></span><span class="op">(</span>tumourname<span class="op">=</span><span class="va">TUMOURNAME</span>,</span>
<span>                                normalname<span class="op">=</span><span class="va">NORMALNAME</span>,</span>
<span>                                tumour_data_file<span class="op">=</span><span class="va">TUMOURBAM</span>,</span>
<span>                                normal_data_file<span class="op">=</span><span class="va">NORMALBAM</span>,</span>
<span>                                imputeinfofile<span class="op">=</span><span class="va">IMPUTEINFOFILE</span>,</span>
<span>                                g1000prefix<span class="op">=</span><span class="va">G1000PREFIX</span>,</span>
<span>                                problemloci<span class="op">=</span><span class="va">PROBLEMLOCI</span>,</span>
<span>                                gccorrectprefix<span class="op">=</span><span class="va">GCCORRECTPREFIX</span>,</span>
<span>                                repliccorrectprefix<span class="op">=</span><span class="va">REPLICCORRECTPREFIX</span>,</span>
<span>                                g1000allelesprefix<span class="op">=</span><span class="va">G1000PREFIX_AC</span>,</span>
<span>                                ismale<span class="op">=</span><span class="va">IS_MALE</span>,</span>
<span>                                data_type<span class="op">=</span><span class="st">"wgs"</span>,</span>
<span>                                impute_exe<span class="op">=</span><span class="st">"impute2"</span>,</span>
<span>                                allelecounter_exe<span class="op">=</span><span class="st">"alleleCounter"</span>,</span>
<span>                                nthreads<span class="op">=</span><span class="va">NTHREADS</span>,</span>
<span>                                platform_gamma<span class="op">=</span><span class="fl">1</span>,</span>
<span>                                phasing_gamma<span class="op">=</span><span class="fl">1</span>,</span>
<span>                                segmentation_gamma<span class="op">=</span><span class="fl">10</span>,</span>
<span>                                segmentation_kmin<span class="op">=</span><span class="fl">3</span>,</span>
<span>                                phasing_kmin<span class="op">=</span><span class="fl">1</span>,</span>
<span>                                clonality_dist_metric<span class="op">=</span><span class="fl">0</span>,</span>
<span>                                ascat_dist_metric<span class="op">=</span><span class="fl">1</span>,</span>
<span>                                min_ploidy<span class="op">=</span><span class="fl">1.6</span>,</span>
<span>                                max_ploidy<span class="op">=</span><span class="fl">4.8</span>, min_rho<span class="op">=</span><span class="fl">0.1</span>,</span>
<span>                                min_goodness<span class="op">=</span><span class="fl">0.63</span>,</span>
<span>                                uninformative_BAF_threshold<span class="op">=</span><span class="fl">0.51</span>,</span>
<span>                                min_normal_depth<span class="op">=</span><span class="fl">10</span>,</span>
<span>                                min_base_qual<span class="op">=</span><span class="fl">20</span>,</span>
<span>                                min_map_qual<span class="op">=</span><span class="fl">35</span>,</span>
<span>                                calc_seg_baf_option<span class="op">=</span><span class="fl">1</span>,</span>
<span>                                skip_allele_counting<span class="op">=</span><span class="cn">F</span>,</span>
<span>                                skip_preprocessing<span class="op">=</span><span class="cn">F</span>,</span>
<span>                                skip_phasing<span class="op">=</span><span class="cn">F</span>,</span>
<span>                                usebeagle<span class="op">=</span><span class="va">USEBEAGLE</span>, <span class="co">##set to TRUE to use beagle</span></span>
<span>                                beaglejar<span class="op">=</span><span class="va">BEAGLEJAR</span>, <span class="co">##path</span></span>
<span>                                beagleref<span class="op">=</span><span class="va">BEAGLEREF.template</span>, <span class="co">##pathtemplate</span></span>
<span>                                beagleplink<span class="op">=</span><span class="va">BEAGLEPLINK.template</span>, <span class="co">##pathtemplate</span></span>
<span>                                beaglemaxmem<span class="op">=</span><span class="fl">15</span>,</span>
<span>                                beaglenthreads<span class="op">=</span><span class="fl">1</span>,</span>
<span>                                beaglewindow<span class="op">=</span><span class="fl">40</span>,</span>
<span>                                beagleoverlap<span class="op">=</span><span class="fl">4</span>,</span>
<span>                                snp6_reference_info_file<span class="op">=</span><span class="cn">NA</span>,</span>
<span>                                apt.probeset.genotype.exe<span class="op">=</span><span class="st">"apt-probeset-genotype"</span>,</span>
<span>                                apt.probeset.summarize.exe<span class="op">=</span><span class="st">"apt-probeset-summarize"</span>,</span>
<span>                                norm.geno.clust.exe<span class="op">=</span><span class="st">"normalize_affy_geno_cluster.pl"</span>,</span>
<span>                                birdseed_report_file<span class="op">=</span><span class="st">"birdseed.report.txt"</span>,</span>
<span>                                heterozygousFilter<span class="op">=</span><span class="st">"none"</span>,</span>
<span>                                prior_breakpoints_file<span class="op">=</span><span class="cn">NULL</span><span class="op">)</span><span class="op">)</span></span></code></pre>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing Battenberg</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>David Wedge <br><small class="roles"> Author </small>   </li>
<li>Peter Van Loo <br><small class="roles"> Author </small>   </li>
<li>Naser Ansari-Pour <br><small class="roles"> Author </small>   </li>
<li>Stefan Dentro <br><small class="roles"> Author, maintainer </small>   </li>
<li>Maxime Tarabichi <br><small class="roles"> Author </small>   </li>
<li>Jonas Demeulemeester <br><small class="roles"> Author </small>   </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer>
</div>






  </body>
</html>
