<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Advanced Usage and Parameter Optimization â€¢ Battenberg</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Advanced Usage and Parameter Optimization">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Advanced Usage and Parameter Optimization</h1>
                        <h4 data-toc-skip class="author">David Wedge Group</h4>
            
            <h4 data-toc-skip class="date">2025-07-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/Wedge-Oxford/battenberg/blob/HEAD/vignettes/advanced-usage.Rmd" class="external-link"><code>vignettes/advanced-usage.Rmd</code></a></small>
      <div class="hidden name"><code>advanced-usage.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Wedge-Oxford/battenberg" class="external-link">Battenberg</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="advanced-parameter-tuning">Advanced Parameter Tuning<a class="anchor" aria-label="anchor" href="#advanced-parameter-tuning"></a>
</h2>
<div class="section level3">
<h3 id="segmentation-parameters">Segmentation Parameters<a class="anchor" aria-label="anchor" href="#segmentation-parameters"></a>
</h3>
<p>The segmentation behavior can be controlled by several parameters:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># More sensitive segmentation (more segments)</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  segmentation_gamma <span class="op">=</span> <span class="fl">25</span>,  <span class="co"># Higher = more segments</span></span>
<span>  segmentation_kmin <span class="op">=</span> <span class="fl">1</span>,    <span class="co"># Minimum segment size</span></span>
<span>  phasing_kmin <span class="op">=</span> <span class="fl">1</span>         <span class="co"># Minimum size for phasing</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Less sensitive segmentation (fewer segments)</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  segmentation_gamma <span class="op">=</span> <span class="fl">5</span>,   <span class="co"># Lower = fewer segments</span></span>
<span>  segmentation_kmin <span class="op">=</span> <span class="fl">5</span>,    <span class="co"># Larger minimum segment size</span></span>
<span>  phasing_kmin <span class="op">=</span> <span class="fl">3</span>         <span class="co"># Larger minimum size for phasing</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="purity-and-ploidy-constraints">Purity and Ploidy Constraints<a class="anchor" aria-label="anchor" href="#purity-and-ploidy-constraints"></a>
</h3>
<p>Adjust expected ranges based on sample characteristics:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># High purity sample</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  min_rho <span class="op">=</span> <span class="fl">0.8</span>,           <span class="co"># Minimum 80% purity</span></span>
<span>  max_ploidy <span class="op">=</span> <span class="fl">6.0</span>,        <span class="co"># Allow higher ploidy</span></span>
<span>  min_goodness <span class="op">=</span> <span class="fl">0.65</span>      <span class="co"># Stricter goodness of fit</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Low purity sample</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  min_rho <span class="op">=</span> <span class="fl">0.3</span>,           <span class="co"># Allow 30% purity</span></span>
<span>  max_ploidy <span class="op">=</span> <span class="fl">3.5</span>,        <span class="co"># Lower ploidy range</span></span>
<span>  min_goodness <span class="op">=</span> <span class="fl">0.55</span>      <span class="co"># More lenient goodness</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quality-control-parameters">Quality Control Parameters<a class="anchor" aria-label="anchor" href="#quality-control-parameters"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Strict quality control</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  min_normal_depth <span class="op">=</span> <span class="fl">15</span>,   <span class="co"># Higher coverage requirement</span></span>
<span>  min_base_qual <span class="op">=</span> <span class="fl">25</span>,      <span class="co"># Higher base quality</span></span>
<span>  min_map_qual <span class="op">=</span> <span class="fl">40</span>,       <span class="co"># Higher mapping quality</span></span>
<span>  uninformative_BAF_threshold <span class="op">=</span> <span class="fl">0.49</span>  <span class="co"># Stricter BAF threshold</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="using-prior-structural-variant-breakpoints">Using Prior Structural Variant Breakpoints<a class="anchor" aria-label="anchor" href="#using-prior-structural-variant-breakpoints"></a>
</h2>
<p>Battenberg can incorporate prior breakpoints from structural variant calls:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create prior breakpoints file (2 columns: chr, pos)</span></span>
<span><span class="va">prior_breakpoints</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span></span>
<span>  chr <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1"</span>, <span class="st">"1"</span>, <span class="st">"2"</span>, <span class="st">"3"</span><span class="op">)</span>,</span>
<span>  pos <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1500000</span>, <span class="fl">2500000</span>, <span class="fl">5000000</span>, <span class="fl">1000000</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">prior_breakpoints</span>, <span class="st">"prior_breakpoints.txt"</span>, </span>
<span>            row.names <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>            quote <span class="op">=</span> <span class="cn">FALSE</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use in Battenberg</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  prior_breakpoints_file <span class="op">=</span> <span class="st">"prior_breakpoints.txt"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="using-beagle5-for-imputation">Using Beagle5 for Imputation<a class="anchor" aria-label="anchor" href="#using-beagle5-for-imputation"></a>
</h2>
<p>For improved phasing, especially with newer reference panels:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Setup Beagle5 parameters</span></span>
<span><span class="va">BEAGLEJAR</span> <span class="op">&lt;-</span> <span class="st">"path/to/beagle.24Aug19.3e8.jar"</span></span>
<span><span class="va">BEAGLEREF_TEMPLATE</span> <span class="op">&lt;-</span> <span class="st">"path/to/beagle_ref_chrCHROMNAME.1kg.phase3.v5a.b37.bref3"</span></span>
<span><span class="va">BEAGLEPLINK_TEMPLATE</span> <span class="op">&lt;-</span> <span class="st">"path/to/plink.chrCHROMNAME.GRCh37.map"</span></span>
<span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  usebeagle <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  beaglejar <span class="op">=</span> <span class="va">BEAGLEJAR</span>,</span>
<span>  beagleref <span class="op">=</span> <span class="va">BEAGLEREF_TEMPLATE</span>,</span>
<span>  beagleplink <span class="op">=</span> <span class="va">BEAGLEPLINK_TEMPLATE</span>,</span>
<span>  beaglemaxmem <span class="op">=</span> <span class="fl">16</span>,       <span class="co"># Memory in GB</span></span>
<span>  beaglenthreads <span class="op">=</span> <span class="fl">4</span>,      <span class="co"># Threads for Beagle</span></span>
<span>  beaglewindow <span class="op">=</span> <span class="fl">40</span>,       <span class="co"># Window size</span></span>
<span>  beagleoverlap <span class="op">=</span> <span class="fl">4</span>        <span class="co"># Overlap size</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="multisample-analysis">Multisample Analysis<a class="anchor" aria-label="anchor" href="#multisample-analysis"></a>
</h2>
<p>For analyzing multiple samples together:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define multiple samples</span></span>
<span><span class="va">tumournames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample1_tumor"</span>, <span class="st">"sample2_tumor"</span>, <span class="st">"sample3_tumor"</span><span class="op">)</span></span>
<span><span class="va">normalnames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sample1_normal"</span>, <span class="st">"sample2_normal"</span>, <span class="st">"sample3_normal"</span><span class="op">)</span></span>
<span><span class="va">tumourbams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"path/to/sample1_tumor.bam"</span>, <span class="st">"path/to/sample2_tumor.bam"</span>, </span>
<span>                <span class="st">"path/to/sample3_tumor.bam"</span><span class="op">)</span></span>
<span><span class="va">normalbams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"path/to/sample1_normal.bam"</span>, <span class="st">"path/to/sample2_normal.bam"</span>, </span>
<span>                <span class="st">"path/to/sample3_normal.bam"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Run multisample analysis</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  tumourname <span class="op">=</span> <span class="va">tumournames</span>,</span>
<span>  normalname <span class="op">=</span> <span class="va">normalnames</span>,</span>
<span>  tumour_data_file <span class="op">=</span> <span class="va">tumourbams</span>,</span>
<span>  normal_data_file <span class="op">=</span> <span class="va">normalbams</span>,</span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  multisample_maxlag <span class="op">=</span> <span class="fl">150</span>,  <span class="co"># Max upstream SNPs for multisample phasing</span></span>
<span>  multisample_relative_weight_balanced <span class="op">=</span> <span class="fl">0.5</span>,  <span class="co"># Weight for balanced samples</span></span>
<span>  write_battenberg_phasing <span class="op">=</span> <span class="cn">TRUE</span>  <span class="co"># Write phasing results</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cell-line-analysis">Cell Line Analysis<a class="anchor" aria-label="anchor" href="#cell-line-analysis"></a>
</h2>
<p>For cell line data (tumor-only analysis):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  analysis <span class="op">=</span> <span class="st">"cell_line"</span>,  <span class="co"># Changed from default "paired"</span></span>
<span>  tumourname <span class="op">=</span> <span class="st">"cell_line_sample"</span>,</span>
<span>  normalname <span class="op">=</span> <span class="cn">NA</span>,         <span class="co"># No normal sample</span></span>
<span>  tumour_data_file <span class="op">=</span> <span class="st">"path/to/cell_line.bam"</span>,</span>
<span>  normal_data_file <span class="op">=</span> <span class="cn">NA</span>,   <span class="co"># No normal BAM</span></span>
<span>  <span class="co"># ... other parameters adjusted for cell line analysis ...</span></span>
<span>  min_rho <span class="op">=</span> <span class="fl">0.95</span>,         <span class="co"># Expect high purity</span></span>
<span>  min_goodness <span class="op">=</span> <span class="fl">0.7</span>      <span class="co"># Stricter goodness for cell lines</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="snp-array-analysis">SNP Array Analysis<a class="anchor" aria-label="anchor" href="#snp-array-analysis"></a>
</h2>
<p>For SNP6 array data:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  data_type <span class="op">=</span> <span class="st">"snp6"</span>,</span>
<span>  platform_gamma <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  snp6_reference_info_file <span class="op">=</span> <span class="st">"path/to/snp6_reference_info.txt"</span>,</span>
<span>  apt.probeset.genotype.exe <span class="op">=</span> <span class="st">"apt-probeset-genotype"</span>,</span>
<span>  apt.probeset.summarize.exe <span class="op">=</span> <span class="st">"apt-probeset-summarize"</span>,</span>
<span>  norm.geno.clust.exe <span class="op">=</span> <span class="st">"normalize_affy_geno_cluster.pl"</span>,</span>
<span>  birdseed_report_file <span class="op">=</span> <span class="st">"birdseed.report.txt"</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="performance-optimization">Performance Optimization<a class="anchor" aria-label="anchor" href="#performance-optimization"></a>
</h2>
<div class="section level3">
<h3 id="parallel-processing">Parallel Processing<a class="anchor" aria-label="anchor" href="#parallel-processing"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use more threads for faster processing</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  nthreads <span class="op">=</span> <span class="fl">16</span>,           <span class="co"># Use 16 CPU cores</span></span>
<span>  beaglenthreads <span class="op">=</span> <span class="fl">8</span>       <span class="co"># Use 8 cores for Beagle (if using)</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="memory-management">Memory Management<a class="anchor" aria-label="anchor" href="#memory-management"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For large datasets, adjust memory settings</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  beaglemaxmem <span class="op">=</span> <span class="fl">32</span>,       <span class="co"># 32GB for Beagle</span></span>
<span>  <span class="co"># Consider running chromosomes separately for very large files</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="skipping-steps">Skipping Steps<a class="anchor" aria-label="anchor" href="#skipping-steps"></a>
</h3>
<p>For rerunning parts of the analysis:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Skip allele counting if already done</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  skip_allele_counting <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  skip_preprocessing <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  skip_phasing <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Skip preprocessing if rerunning</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  skip_allele_counting <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  skip_preprocessing <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  skip_phasing <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="custom-genome-builds">Custom Genome Builds<a class="anchor" aria-label="anchor" href="#custom-genome-builds"></a>
</h2>
<p>For different reference genomes:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Specify genome build</span></span>
<span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  GENOMEBUILD <span class="op">=</span> <span class="st">"hg38"</span>,  <span class="co"># or "hg19"</span></span>
<span>  <span class="co"># Ensure reference files match the specified build</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="external-haplotype-files">External Haplotype Files<a class="anchor" aria-label="anchor" href="#external-haplotype-files"></a>
</h2>
<p>Using external phasing information:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  <span class="co"># ... other parameters ...</span></span>
<span>  externalhaplotypefile <span class="op">=</span> <span class="st">"path/to/external_haplotypes.vcf"</span>,</span>
<span>  write_battenberg_phasing <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="troubleshooting-common-issues">Troubleshooting Common Issues<a class="anchor" aria-label="anchor" href="#troubleshooting-common-issues"></a>
</h2>
<div class="section level3">
<h3 id="low-quality-samples">Low Quality Samples<a class="anchor" aria-label="anchor" href="#low-quality-samples"></a>
</h3>
<ul>
<li>Increase <code>min_normal_depth</code> and quality thresholds</li>
<li>Adjust <code>min_goodness</code> to be more lenient</li>
<li>Check coverage uniformity</li>
</ul>
</div>
<div class="section level3">
<h3 id="highly-aneuploid-samples">Highly Aneuploid Samples<a class="anchor" aria-label="anchor" href="#highly-aneuploid-samples"></a>
</h3>
<ul>
<li>Increase <code>max_ploidy</code> range</li>
<li>Adjust <code>segmentation_gamma</code> for appropriate segment resolution</li>
</ul>
</div>
<div class="section level3">
<h3 id="contaminated-samples">Contaminated Samples<a class="anchor" aria-label="anchor" href="#contaminated-samples"></a>
</h3>
<ul>
<li>Lower <code>min_rho</code> threshold</li>
<li>Consider pre-processing to estimate contamination</li>
</ul>
</div>
<div class="section level3">
<h3 id="memory-issues">Memory Issues<a class="anchor" aria-label="anchor" href="#memory-issues"></a>
</h3>
<ul>
<li>Reduce <code>beaglemaxmem</code> if running out of memory</li>
<li>Process chromosomes separately</li>
<li>Use fewer threads if memory-limited</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="quality-assessment">Quality Assessment<a class="anchor" aria-label="anchor" href="#quality-assessment"></a>
</h2>
<p>After running Battenberg, assess quality using:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Distance plot</strong>: Check purity/ploidy solution space</li>
<li>
<strong>Profile plots</strong>: Examine copy number profiles for artifacts</li>
<li>
<strong>Coverage plots</strong>: Verify uniform coverage</li>
<li>
<strong>BAF plots</strong>: Check for proper phase separation</li>
</ol>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example quality check</span></span>
<span><span class="va">cn_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"sample_tumor_copynumber.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check for very short segments (potential artifacts)</span></span>
<span><span class="va">short_segments</span> <span class="op">&lt;-</span> <span class="va">cn_data</span><span class="op">[</span><span class="va">cn_data</span><span class="op">$</span><span class="va">endpos</span> <span class="op">-</span> <span class="va">cn_data</span><span class="op">$</span><span class="va">startpos</span> <span class="op">&lt;</span> <span class="fl">1000000</span>, <span class="op">]</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">short_segments</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Warning: Found"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">short_segments</span><span class="op">)</span>, <span class="st">"segments &lt; 1Mb\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Check purity estimate</span></span>
<span><span class="va">rho_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"sample_tumor_rho_and_psi.txt"</span><span class="op">)</span></span>
<span><span class="va">purity</span> <span class="op">&lt;-</span> <span class="va">rho_psi</span><span class="op">$</span><span class="va">rho</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="kw">if</span><span class="op">(</span><span class="va">purity</span> <span class="op">&lt;</span> <span class="fl">0.3</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Warning: Low estimated purity:"</span>, <span class="va">purity</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer>
</div>






  </body>
</html>
