<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting Started with Battenberg • Battenberg</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with Battenberg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting Started with Battenberg</h1>
                        <h4 data-toc-skip class="author">David Wedge Group</h4>
            
            <h4 data-toc-skip class="date">2025-07-04</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/Wedge-Oxford/battenberg/blob/HEAD/vignettes/getting-started.Rmd" class="external-link"><code>vignettes/getting-started.Rmd</code></a></small>
      <div class="hidden name"><code>getting-started.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Wedge-Oxford/battenberg" class="external-link">Battenberg</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Battenberg is a whole genome sequencing subclonal copy number caller that estimates subclonal copy number alterations from matched tumor-normal whole genome sequencing data. It can detect both clonal and subclonal copy number changes and provides estimates of tumor purity and ploidy.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="prerequisites">Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h3>
<p>Battenberg requires several dependencies. Install them first:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install Bioconductor packages</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"devtools"</span>, <span class="st">"splines"</span>, <span class="st">"readr"</span>, <span class="st">"doParallel"</span>, </span>
<span>                       <span class="st">"ggplot2"</span>, <span class="st">"RColorBrewer"</span>, <span class="st">"gridExtra"</span>, <span class="st">"gtools"</span>, </span>
<span>                       <span class="st">"parallel"</span>, <span class="st">"VariantAnnotation"</span>, <span class="st">"GenomicRanges"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install modified copynumber package</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"igordot/copynumber"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Install ASCAT</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"VanLoo-lab/ascat/ASCAT"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="install-battenberg">Install Battenberg<a class="anchor" aria-label="anchor" href="#install-battenberg"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from GitHub (pre_3.0 branch)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"Wedge-lab/battenberg"</span>, ref<span class="op">=</span><span class="st">"pre_3.0"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reference-data-requirements">Reference Data Requirements<a class="anchor" aria-label="anchor" href="#reference-data-requirements"></a>
</h2>
<p>Before running Battenberg, you need to download reference data:</p>
<div class="section level3">
<h3 id="for-grch37hg19">For GRCh37/hg19:<a class="anchor" aria-label="anchor" href="#for-grch37hg19"></a>
</h3>
<ul>
<li>Download from: <a href="https://ora.ox.ac.uk/objects/uuid:2c1fec09-a504-49ab-9ce9-3f17bac531bc" class="external-link uri">https://ora.ox.ac.uk/objects/uuid:2c1fec09-a504-49ab-9ce9-3f17bac531bc</a>
</li>
<li>Files include: 1000G loci, imputation reference, GC correction, replication timing correction</li>
</ul>
</div>
<div class="section level3">
<h3 id="for-grch38hg38">For GRCh38/hg38:<a class="anchor" aria-label="anchor" href="#for-grch38hg38"></a>
</h3>
<ul>
<li>Download from: <a href="https://ora.ox.ac.uk/objects/uuid:08e24957-7e76-438a-bd38-66c48008cf52" class="external-link uri">https://ora.ox.ac.uk/objects/uuid:08e24957-7e76-438a-bd38-66c48008cf52</a>
</li>
<li>Files include: 1000G loci, imputation reference, GC correction, replication timing correction</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="basic-usage">Basic Usage<a class="anchor" aria-label="anchor" href="#basic-usage"></a>
</h2>
<div class="section level3">
<h3 id="running-the-full-pipeline">Running the Full Pipeline<a class="anchor" aria-label="anchor" href="#running-the-full-pipeline"></a>
</h3>
<p>The main function <code><a href="../reference/battenberg.html">battenberg()</a></code> runs the complete analysis pipeline:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define sample names and file paths</span></span>
<span><span class="va">TUMOURNAME</span> <span class="op">&lt;-</span> <span class="st">"sample_tumor"</span></span>
<span><span class="va">NORMALNAME</span> <span class="op">&lt;-</span> <span class="st">"sample_normal"</span></span>
<span><span class="va">TUMOURBAM</span> <span class="op">&lt;-</span> <span class="st">"path/to/tumor.bam"</span></span>
<span><span class="va">NORMALBAM</span> <span class="op">&lt;-</span> <span class="st">"path/to/normal.bam"</span></span>
<span></span>
<span><span class="co"># Reference file paths (adjust to your setup)</span></span>
<span><span class="va">IMPUTEINFOFILE</span> <span class="op">&lt;-</span> <span class="st">"path/to/impute_info.txt"</span></span>
<span><span class="va">G1000PREFIX</span> <span class="op">&lt;-</span> <span class="st">"path/to/1000genomes/prefix"</span></span>
<span><span class="va">PROBLEMLOCI</span> <span class="op">&lt;-</span> <span class="st">"path/to/probloci.txt"</span></span>
<span><span class="va">GCCORRECTPREFIX</span> <span class="op">&lt;-</span> <span class="st">"path/to/gc_correction/prefix"</span></span>
<span><span class="va">REPLICCORRECTPREFIX</span> <span class="op">&lt;-</span> <span class="st">"path/to/replic_correction/prefix"</span></span>
<span><span class="va">G1000PREFIX_AC</span> <span class="op">&lt;-</span> <span class="st">"path/to/1000genomes_alleles/prefix"</span></span>
<span></span>
<span><span class="co"># Determine if sample is male (TRUE) or female (FALSE)</span></span>
<span><span class="va">IS_MALE</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span></span>
<span><span class="co"># Number of threads to use</span></span>
<span><span class="va">NTHREADS</span> <span class="op">&lt;-</span> <span class="fl">8</span></span>
<span></span>
<span><span class="co"># Run Battenberg</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/battenberg.html">battenberg</a></span><span class="op">(</span></span>
<span>  tumourname <span class="op">=</span> <span class="va">TUMOURNAME</span>,</span>
<span>  normalname <span class="op">=</span> <span class="va">NORMALNAME</span>,</span>
<span>  tumour_data_file <span class="op">=</span> <span class="va">TUMOURBAM</span>,</span>
<span>  normal_data_file <span class="op">=</span> <span class="va">NORMALBAM</span>,</span>
<span>  imputeinfofile <span class="op">=</span> <span class="va">IMPUTEINFOFILE</span>,</span>
<span>  g1000prefix <span class="op">=</span> <span class="va">G1000PREFIX</span>,</span>
<span>  problemloci <span class="op">=</span> <span class="va">PROBLEMLOCI</span>,</span>
<span>  gccorrectprefix <span class="op">=</span> <span class="va">GCCORRECTPREFIX</span>,</span>
<span>  repliccorrectprefix <span class="op">=</span> <span class="va">REPLICCORRECTPREFIX</span>,</span>
<span>  g1000allelesprefix <span class="op">=</span> <span class="va">G1000PREFIX_AC</span>,</span>
<span>  ismale <span class="op">=</span> <span class="va">IS_MALE</span>,</span>
<span>  data_type <span class="op">=</span> <span class="st">"wgs"</span>,</span>
<span>  nthreads <span class="op">=</span> <span class="va">NTHREADS</span>,</span>
<span>  platform_gamma <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  phasing_gamma <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  segmentation_gamma <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  segmentation_kmin <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  phasing_kmin <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  min_ploidy <span class="op">=</span> <span class="fl">1.6</span>,</span>
<span>  max_ploidy <span class="op">=</span> <span class="fl">4.8</span>,</span>
<span>  min_rho <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  min_goodness <span class="op">=</span> <span class="fl">0.63</span>,</span>
<span>  uninformative_BAF_threshold <span class="op">=</span> <span class="fl">0.51</span>,</span>
<span>  min_normal_depth <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  min_base_qual <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  min_map_qual <span class="op">=</span> <span class="fl">35</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="key-parameters">Key Parameters<a class="anchor" aria-label="anchor" href="#key-parameters"></a>
</h3>
<ul>
<li>
<strong>tumourname/normalname</strong>: Sample identifiers used as prefixes for output files</li>
<li>
<strong>tumour_data_file/normal_data_file</strong>: Paths to BAM files</li>
<li>
<strong>data_type</strong>: “wgs” for whole genome sequencing, “snp6” for SNP array data</li>
<li>
<strong>ismale</strong>: TRUE for male samples, FALSE for female samples</li>
<li>
<strong>platform_gamma</strong>: Platform-specific gamma parameter (1 for WGS, 1 for SNP6)</li>
<li>
<strong>segmentation_gamma</strong>: Controls segmentation sensitivity (higher = more segments)</li>
<li>
<strong>min_ploidy/max_ploidy</strong>: Expected range of tumor ploidy</li>
<li>
<strong>min_rho</strong>: Minimum tumor purity to consider</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="output-files">Output Files<a class="anchor" aria-label="anchor" href="#output-files"></a>
</h2>
<p>Battenberg produces several key output files:</p>
<div class="section level3">
<h3 id="output-files-1">Output Files<a class="anchor" aria-label="anchor" href="#output-files-1"></a>
</h3>
<p>Battenberg produces several key output files:</p>
</div>
<div class="section level3">
<h3 id="primary-results">Primary Results<a class="anchor" aria-label="anchor" href="#primary-results"></a>
</h3>
<ul>
<li>
<code>[samplename]_copynumber.txt</code>: Copy number segments with clonal/subclonal states</li>
<li>
<code>[samplename]_rho_and_psi.txt</code>: Tumor purity and ploidy estimates</li>
</ul>
</div>
<div class="section level3">
<h3 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h3>
<ul>
<li>
<code>[samplename]_BattenbergProfile.png</code>: Genome-wide copy number profile</li>
<li>
<code>[samplename]_BattenbergProfile_subclones.png</code>: Alternative subclonal view</li>
<li>
<code>[samplename]_subclones_chr*.png</code>: Per-chromosome detailed plots</li>
<li>
<code>[samplename]_distance.png</code>: Purity/ploidy solution space</li>
</ul>
</div>
<div class="section level3">
<h3 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h3>
<ul>
<li>
<code>[samplename].tumour.png</code>: Raw tumor BAF and LogR</li>
<li>
<code>[samplename].germline.png</code>: Raw normal BAF and LogR</li>
<li>
<code>[samplename]_coverage.png</code>: Coverage profiles</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="reading-results">Reading Results<a class="anchor" aria-label="anchor" href="#reading-results"></a>
</h2>
<div class="section level3">
<h3 id="load-copy-number-data">Load Copy Number Data<a class="anchor" aria-label="anchor" href="#load-copy-number-data"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read the main results file</span></span>
<span><span class="va">cn_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"sample_tumor_copynumber.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Examine the structure</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cn_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">cn_data</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-purityploidy-estimates">Load Purity/Ploidy Estimates<a class="anchor" aria-label="anchor" href="#load-purityploidy-estimates"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Read purity and ploidy estimates</span></span>
<span><span class="va">rho_psi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="st">"sample_tumor_rho_and_psi.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Extract purity (rho) - use FRAC_genome value from second row</span></span>
<span><span class="va">tumor_purity</span> <span class="op">&lt;-</span> <span class="va">rho_psi</span><span class="op">$</span><span class="va">rho</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">tumor_ploidy</span> <span class="op">&lt;-</span> <span class="va">rho_psi</span><span class="op">$</span><span class="va">psi</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Estimated tumor purity:"</span>, <span class="va">tumor_purity</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Estimated tumor ploidy:"</span>, <span class="va">tumor_ploidy</span>, <span class="st">"\n"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="understanding-the-output">Understanding the Output<a class="anchor" aria-label="anchor" href="#understanding-the-output"></a>
</h2>
<div class="section level3">
<h3 id="copy-number-states">Copy Number States<a class="anchor" aria-label="anchor" href="#copy-number-states"></a>
</h3>
<p>Each segment can have: - <strong>Clonal</strong>: Single copy number state (frac1_A = 1, frac2_A = NA) - <strong>Subclonal</strong>: Two copy number states (frac1_A + frac2_A = 1)</p>
</div>
<div class="section level3">
<h3 id="key-columns-in-copynumber-txt">Key Columns in copynumber.txt<a class="anchor" aria-label="anchor" href="#key-columns-in-copynumber-txt"></a>
</h3>
<ul>
<li>
<code>nMaj1_A</code>, <code>nMin1_A</code>: Major/minor allele copy numbers for state 1</li>
<li>
<code>nMaj2_A</code>, <code>nMin2_A</code>: Major/minor allele copy numbers for state 2 (if subclonal)</li>
<li>
<code>frac1_A</code>, <code>frac2_A</code>: Fraction of tumor cells with each state</li>
<li>
<code>pval</code>: P-value for subclonal vs clonal model</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="next-steps">Next Steps<a class="anchor" aria-label="anchor" href="#next-steps"></a>
</h2>
<ul>
<li>Read <a href="advanced-usage.html">Advanced Usage</a> for parameter optimization</li>
<li>See <a href="data-interpretation.html">Data Interpretation</a> for result analysis</li>
<li>Check <a href="troubleshooting.html">Troubleshooting</a> for common issues</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer>
</div>






  </body>
</html>
