<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Run the Battenberg pipeline — battenberg • Battenberg</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Run the Battenberg pipeline — battenberg"><meta property="og:description" content="Run the Battenberg pipeline"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Run the Battenberg pipeline</h1>
    <small class="dont-index">Source: <a href="https://github.com/Wedge-Oxford/battenberg/blob/HEAD/R/battenberg.R" class="external-link"><code>R/battenberg.R</code></a></small>
    <div class="hidden name"><code>battenberg.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Run the Battenberg pipeline</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">battenberg</span><span class="op">(</span></span>
<span>  analysis <span class="op">=</span> <span class="st">"paired"</span>,</span>
<span>  <span class="va">samplename</span>,</span>
<span>  <span class="va">normalname</span>,</span>
<span>  <span class="va">sample_data_file</span>,</span>
<span>  <span class="va">normal_data_file</span>,</span>
<span>  <span class="va">imputeinfofile</span>,</span>
<span>  <span class="va">g1000prefix</span>,</span>
<span>  <span class="va">problemloci</span>,</span>
<span>  gccorrectprefix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  repliccorrectprefix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  g1000allelesprefix <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  ismale <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  data_type <span class="op">=</span> <span class="st">"wgs"</span>,</span>
<span>  impute_exe <span class="op">=</span> <span class="st">"impute2"</span>,</span>
<span>  allelecounter_exe <span class="op">=</span> <span class="st">"alleleCounter"</span>,</span>
<span>  nthreads <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  platform_gamma <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  phasing_gamma <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  segmentation_gamma <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  segmentation_kmin <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  phasing_kmin <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  clonality_dist_metric <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  ascat_dist_metric <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  min_ploidy <span class="op">=</span> <span class="fl">1.6</span>,</span>
<span>  max_ploidy <span class="op">=</span> <span class="fl">4.8</span>,</span>
<span>  min_rho <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  max_rho <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  min_goodness <span class="op">=</span> <span class="fl">0.63</span>,</span>
<span>  uninformative_BAF_threshold <span class="op">=</span> <span class="fl">0.51</span>,</span>
<span>  min_normal_depth <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  min_base_qual <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  min_map_qual <span class="op">=</span> <span class="fl">35</span>,</span>
<span>  max_allowed_state <span class="op">=</span> <span class="fl">250</span>,</span>
<span>  cn_upper_limit <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  calc_seg_baf_option <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  skip_allele_counting <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  skip_preprocessing <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  skip_phasing <span class="op">=</span> <span class="cn">F</span>,</span>
<span>  externalhaplotypefile <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  usebeagle <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  beaglejar <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  beagleref.template <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  beagleplink.template <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  beaglemaxmem <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  beaglenthreads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  beaglewindow <span class="op">=</span> <span class="fl">40</span>,</span>
<span>  beagleoverlap <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  javajre <span class="op">=</span> <span class="st">"java"</span>,</span>
<span>  write_battenberg_phasing <span class="op">=</span> <span class="cn">T</span>,</span>
<span>  multisample_relative_weight_balanced <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  multisample_maxlag <span class="op">=</span> <span class="fl">90</span>,</span>
<span>  segmentation_gamma_multisample <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  snp6_reference_info_file <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  apt.probeset.genotype.exe <span class="op">=</span> <span class="st">"apt-probeset-genotype"</span>,</span>
<span>  apt.probeset.summarize.exe <span class="op">=</span> <span class="st">"apt-probeset-summarize"</span>,</span>
<span>  norm.geno.clust.exe <span class="op">=</span> <span class="st">"normalize_affy_geno_cluster.pl"</span>,</span>
<span>  birdseed_report_file <span class="op">=</span> <span class="st">"birdseed.report.txt"</span>,</span>
<span>  heterozygousFilter <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  prior_breakpoints_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genomebuild <span class="op">=</span> <span class="st">"hg19"</span>,</span>
<span>  chrom_coord_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  enhanced_grid_search <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-analysis">analysis<a class="anchor" aria-label="anchor" href="#arg-analysis"></a></dt>
<dd><p>The mode of Battenberg copy number analysis to be undertaken: 'paired' for tumour-normal pair, 'cell_line' for Cell line tumour-only and 'germline' for germline CNV of normal sample (Default: 'paired')</p></dd>


<dt id="arg-samplename">samplename<a class="anchor" aria-label="anchor" href="#arg-samplename"></a></dt>
<dd><p>Sample identifier (tumour or germline), this is used as a prefix for the output files. If allele counts are supplied separately, they are expected to have this identifier as prefix.</p></dd>


<dt id="arg-normalname">normalname<a class="anchor" aria-label="anchor" href="#arg-normalname"></a></dt>
<dd><p>Matched normal identifier, this is used as a prefix for the output files. If allele counts are supplied separately, they are expected to have this identifier as prefix.</p></dd>


<dt id="arg-sample-data-file">sample_data_file<a class="anchor" aria-label="anchor" href="#arg-sample-data-file"></a></dt>
<dd><p>A BAM or CEL file for the sample</p></dd>


<dt id="arg-normal-data-file">normal_data_file<a class="anchor" aria-label="anchor" href="#arg-normal-data-file"></a></dt>
<dd><p>A BAM or CEL file for the normal-pair (paired analysis)</p></dd>


<dt id="arg-imputeinfofile">imputeinfofile<a class="anchor" aria-label="anchor" href="#arg-imputeinfofile"></a></dt>
<dd><p>Full path to a Battenberg impute info file with pointers to Impute2 reference data</p></dd>


<dt id="arg-g-prefix">g1000prefix<a class="anchor" aria-label="anchor" href="#arg-g-prefix"></a></dt>
<dd><p>Full prefix path to 1000 Genomes SNP loci data, as part of the Battenberg reference data</p></dd>


<dt id="arg-problemloci">problemloci<a class="anchor" aria-label="anchor" href="#arg-problemloci"></a></dt>
<dd><p>Full path to a problem loci file that contains SNP loci that should be filtered out</p></dd>


<dt id="arg-gccorrectprefix">gccorrectprefix<a class="anchor" aria-label="anchor" href="#arg-gccorrectprefix"></a></dt>
<dd><p>Full prefix path to GC content files, as part of the Battenberg reference data, not required for SNP6 data (Default: NULL)</p></dd>


<dt id="arg-repliccorrectprefix">repliccorrectprefix<a class="anchor" aria-label="anchor" href="#arg-repliccorrectprefix"></a></dt>
<dd><p>Full prefix path to replication timing files, as part of the Battenberg reference data, not required for SNP6 data (Default: NULL)</p></dd>


<dt id="arg-g-allelesprefix">g1000allelesprefix<a class="anchor" aria-label="anchor" href="#arg-g-allelesprefix"></a></dt>
<dd><p>Full prefix path to 1000 Genomes SNP alleles data, as part of the Battenberg reference data, not required for SNP6 data (Default: NA)</p></dd>


<dt id="arg-ismale">ismale<a class="anchor" aria-label="anchor" href="#arg-ismale"></a></dt>
<dd><p>A boolean set to TRUE if the donor is male, set to FALSE if female, not required for SNP6 data (Default: NA)</p></dd>


<dt id="arg-data-type">data_type<a class="anchor" aria-label="anchor" href="#arg-data-type"></a></dt>
<dd><p>String that contains either wgs or snp6 depending on the supplied input data (Default: wgs)</p></dd>


<dt id="arg-impute-exe">impute_exe<a class="anchor" aria-label="anchor" href="#arg-impute-exe"></a></dt>
<dd><p>Pointer to the Impute2 executable (Default: impute2, i.e. expected in $PATH)</p></dd>


<dt id="arg-allelecounter-exe">allelecounter_exe<a class="anchor" aria-label="anchor" href="#arg-allelecounter-exe"></a></dt>
<dd><p>Pointer to the alleleCounter executable (Default: alleleCounter, i.e. expected in $PATH)</p></dd>


<dt id="arg-nthreads">nthreads<a class="anchor" aria-label="anchor" href="#arg-nthreads"></a></dt>
<dd><p>The number of concurrent processes to use while running the Battenberg pipeline (Default: 8)</p></dd>


<dt id="arg-platform-gamma">platform_gamma<a class="anchor" aria-label="anchor" href="#arg-platform-gamma"></a></dt>
<dd><p>Platform scaling factor, suggestions are set to 1 for wgs and to 0.55 for snp6 (Default: 1)</p></dd>


<dt id="arg-phasing-gamma">phasing_gamma<a class="anchor" aria-label="anchor" href="#arg-phasing-gamma"></a></dt>
<dd><p>Gamma parameter used when correcting phasing mistakes (Default: 1)</p></dd>


<dt id="arg-segmentation-gamma">segmentation_gamma<a class="anchor" aria-label="anchor" href="#arg-segmentation-gamma"></a></dt>
<dd><p>The gamma parameter controls the size of the penalty of starting a new segment during segmentation. It is therefore the key parameter for controlling the number of segments (Default: 10)</p></dd>


<dt id="arg-segmentation-kmin">segmentation_kmin<a class="anchor" aria-label="anchor" href="#arg-segmentation-kmin"></a></dt>
<dd><p>Kmin represents the minimum number of probes/SNPs that a segment should consist of (Default: 3)</p></dd>


<dt id="arg-phasing-kmin">phasing_kmin<a class="anchor" aria-label="anchor" href="#arg-phasing-kmin"></a></dt>
<dd><p>Kmin used when correcting for phasing mistakes (Default: 3)</p></dd>


<dt id="arg-clonality-dist-metric">clonality_dist_metric<a class="anchor" aria-label="anchor" href="#arg-clonality-dist-metric"></a></dt>
<dd><p>Distance metric to use when choosing purity/ploidy combinations (Default: 0)</p></dd>


<dt id="arg-ascat-dist-metric">ascat_dist_metric<a class="anchor" aria-label="anchor" href="#arg-ascat-dist-metric"></a></dt>
<dd><p>Distance metric to use when choosing purity/ploidy combinations (Default: 1)</p></dd>


<dt id="arg-min-ploidy">min_ploidy<a class="anchor" aria-label="anchor" href="#arg-min-ploidy"></a></dt>
<dd><p>Minimum ploidy to be considered (Default: 1.6)</p></dd>


<dt id="arg-max-ploidy">max_ploidy<a class="anchor" aria-label="anchor" href="#arg-max-ploidy"></a></dt>
<dd><p>Maximum ploidy to be considered (Default: 4.8)</p></dd>


<dt id="arg-min-rho">min_rho<a class="anchor" aria-label="anchor" href="#arg-min-rho"></a></dt>
<dd><p>Minimum purity to be considered (Default: 0.1)</p></dd>


<dt id="arg-max-rho">max_rho<a class="anchor" aria-label="anchor" href="#arg-max-rho"></a></dt>
<dd><p>Maximum purity to be considered (Default: 1.0)</p></dd>


<dt id="arg-min-goodness">min_goodness<a class="anchor" aria-label="anchor" href="#arg-min-goodness"></a></dt>
<dd><p>Minimum goodness of fit required for a purity/ploidy combination to be accepted as a solution (Default: 0.63)</p></dd>


<dt id="arg-uninformative-baf-threshold">uninformative_BAF_threshold<a class="anchor" aria-label="anchor" href="#arg-uninformative-baf-threshold"></a></dt>
<dd><p>The threshold beyond which BAF becomes uninformative (Default: 0.51)</p></dd>


<dt id="arg-min-normal-depth">min_normal_depth<a class="anchor" aria-label="anchor" href="#arg-min-normal-depth"></a></dt>
<dd><p>Minimum depth required in the matched normal for a SNP to be considered as part of the wgs analysis (Default: 10)</p></dd>


<dt id="arg-min-base-qual">min_base_qual<a class="anchor" aria-label="anchor" href="#arg-min-base-qual"></a></dt>
<dd><p>Minimum base quality required for a read to be counted when allele counting (Default: 20)</p></dd>


<dt id="arg-min-map-qual">min_map_qual<a class="anchor" aria-label="anchor" href="#arg-min-map-qual"></a></dt>
<dd><p>Minimum mapping quality required for a read to be counted when allele counting (Default: 35)</p></dd>


<dt id="arg-max-allowed-state">max_allowed_state<a class="anchor" aria-label="anchor" href="#arg-max-allowed-state"></a></dt>
<dd><p>The maximum CN state allowed (Default 250)</p></dd>


<dt id="arg-cn-upper-limit">cn_upper_limit<a class="anchor" aria-label="anchor" href="#arg-cn-upper-limit"></a></dt>
<dd><p>Maximum number of copy number that can be called (Default 1000)</p></dd>


<dt id="arg-calc-seg-baf-option">calc_seg_baf_option<a class="anchor" aria-label="anchor" href="#arg-calc-seg-baf-option"></a></dt>
<dd><p>Sets way to calculate BAF per segment: 1=mean, 2=median, 3=ifelse median==0 | 1, mean, median (Default (paired): 3, cell_line &amp; germline: 1)</p></dd>


<dt id="arg-skip-allele-counting">skip_allele_counting<a class="anchor" aria-label="anchor" href="#arg-skip-allele-counting"></a></dt>
<dd><p>Provide TRUE when allele counting can be skipped (i.e. its already done) (Default: FALSE)</p></dd>


<dt id="arg-skip-preprocessing">skip_preprocessing<a class="anchor" aria-label="anchor" href="#arg-skip-preprocessing"></a></dt>
<dd><p>Provide TRUE when preprocessing is already complete (Default: FALSE)</p></dd>


<dt id="arg-skip-phasing">skip_phasing<a class="anchor" aria-label="anchor" href="#arg-skip-phasing"></a></dt>
<dd><p>Provide TRUE when phasing is already complete (Default: FALSE)</p></dd>


<dt id="arg-externalhaplotypefile">externalhaplotypefile<a class="anchor" aria-label="anchor" href="#arg-externalhaplotypefile"></a></dt>
<dd><p>Vcf containing externally obtained haplotype blocks (Default: NA)</p></dd>


<dt id="arg-usebeagle">usebeagle<a class="anchor" aria-label="anchor" href="#arg-usebeagle"></a></dt>
<dd><p>Should use beagle5 instead of impute2 Default: FALSE</p></dd>


<dt id="arg-beaglejar">beaglejar<a class="anchor" aria-label="anchor" href="#arg-beaglejar"></a></dt>
<dd><p>Full path to Beagle java jar file Default: NA</p></dd>


<dt id="arg-beagleref-template">beagleref.template<a class="anchor" aria-label="anchor" href="#arg-beagleref-template"></a></dt>
<dd><p>Full path template to Beagle reference files where the chromosome is replaced by 'CHROMNAME' Default: NA</p></dd>


<dt id="arg-beagleplink-template">beagleplink.template<a class="anchor" aria-label="anchor" href="#arg-beagleplink-template"></a></dt>
<dd><p>Full path template to Beagle plink files where the chromosome is replaced by 'CHROMNAME' Default: NA</p></dd>


<dt id="arg-beaglemaxmem">beaglemaxmem<a class="anchor" aria-label="anchor" href="#arg-beaglemaxmem"></a></dt>
<dd><p>Integer Beagle max heap size in Gb  Default: 10</p></dd>


<dt id="arg-beaglenthreads">beaglenthreads<a class="anchor" aria-label="anchor" href="#arg-beaglenthreads"></a></dt>
<dd><p>Integer number of threads used by beagle5 Default:1</p></dd>


<dt id="arg-beaglewindow">beaglewindow<a class="anchor" aria-label="anchor" href="#arg-beaglewindow"></a></dt>
<dd><p>Integer size of the genomic window for beagle5 (cM) Default:40</p></dd>


<dt id="arg-beagleoverlap">beagleoverlap<a class="anchor" aria-label="anchor" href="#arg-beagleoverlap"></a></dt>
<dd><p>Integer size of the overlap between windows beagle5 Default:4</p></dd>


<dt id="arg-javajre">javajre<a class="anchor" aria-label="anchor" href="#arg-javajre"></a></dt>
<dd><p>Path to the Java JRE executable, only required for haplotype reconstruction with Beagle (default java, i.e. in $PATH)</p></dd>


<dt id="arg-write-battenberg-phasing">write_battenberg_phasing<a class="anchor" aria-label="anchor" href="#arg-write-battenberg-phasing"></a></dt>
<dd><p>Write the Battenberg phasing results as vcf to disk, e.g. for multisample cases (Default: TRUE)</p></dd>


<dt id="arg-multisample-relative-weight-balanced">multisample_relative_weight_balanced<a class="anchor" aria-label="anchor" href="#arg-multisample-relative-weight-balanced"></a></dt>
<dd><p>Relative weight to give to haplotype info from a sample without allelic imbalance in the region (Default: 0.25)</p></dd>


<dt id="arg-multisample-maxlag">multisample_maxlag<a class="anchor" aria-label="anchor" href="#arg-multisample-maxlag"></a></dt>
<dd><p>Maximal number of upstream SNPs used in the multisample haplotyping to inform the haplotype at another SNP (Default: 100)</p></dd>


<dt id="arg-segmentation-gamma-multisample">segmentation_gamma_multisample<a class="anchor" aria-label="anchor" href="#arg-segmentation-gamma-multisample"></a></dt>
<dd><p>The gamma parameter controls the size of the penalty of starting a new segment during mutlisample segmentation. It is the key parameter for controlling the number of segments (Default: 10)</p></dd>


<dt id="arg-snp-reference-info-file">snp6_reference_info_file<a class="anchor" aria-label="anchor" href="#arg-snp-reference-info-file"></a></dt>
<dd><p>Reference files for the SNP6 pipeline only (Default: NA)</p></dd>


<dt id="arg-apt-probeset-genotype-exe">apt.probeset.genotype.exe<a class="anchor" aria-label="anchor" href="#arg-apt-probeset-genotype-exe"></a></dt>
<dd><p>Helper tool for extracting data from CEL files, SNP6 pipeline only (Default: apt-probeset-genotype)</p></dd>


<dt id="arg-apt-probeset-summarize-exe">apt.probeset.summarize.exe<a class="anchor" aria-label="anchor" href="#arg-apt-probeset-summarize-exe"></a></dt>
<dd><p>Helper tool for extracting data from CEL files, SNP6 pipeline only (Default: apt-probeset-summarize)</p></dd>


<dt id="arg-norm-geno-clust-exe">norm.geno.clust.exe<a class="anchor" aria-label="anchor" href="#arg-norm-geno-clust-exe"></a></dt>
<dd><p>Helper tool for extracting data from CEL files, SNP6 pipeline only (Default: normalize_affy_geno_cluster.pl)</p></dd>


<dt id="arg-birdseed-report-file">birdseed_report_file<a class="anchor" aria-label="anchor" href="#arg-birdseed-report-file"></a></dt>
<dd><p>Sex inference output file, SNP6 pipeline only (Default: birdseed.report.txt)</p></dd>


<dt id="arg-heterozygousfilter">heterozygousFilter<a class="anchor" aria-label="anchor" href="#arg-heterozygousfilter"></a></dt>
<dd><p>Legacy option to set a heterozygous SNP filter, SNP6 pipeline only (Default: "none")</p></dd>


<dt id="arg-prior-breakpoints-file">prior_breakpoints_file<a class="anchor" aria-label="anchor" href="#arg-prior-breakpoints-file"></a></dt>
<dd><p>A two column file with prior breakpoints to be used during segmentation (Default: NULL)</p></dd>


<dt id="arg-genomebuild">genomebuild<a class="anchor" aria-label="anchor" href="#arg-genomebuild"></a></dt>
<dd><p>Genome build upon which the 1000G SNP coordinates were obtained (Default: hg19; options: "hg19" or "hg38")</p></dd>


<dt id="arg-enhanced-grid-search">enhanced_grid_search<a class="anchor" aria-label="anchor" href="#arg-enhanced-grid-search"></a></dt>
<dd><p>Should use multi-start, parallelized and multi-approach grid search (Default: FALSE)</p></dd>

</dl></div>
    <div id="author">
    <h2>Author</h2>
    <p>sd11, jdemeul, Naser Ansari-Pour, Julio Cesar Cortes Rios</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

