<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Reconstruct normal-pair allele count files for Germlines — germline_reconstruct_normal • Battenberg</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Reconstruct normal-pair allele count files for Germlines — germline_reconstruct_normal"><meta property="og:description" content="Function to generate normal-pair allele count files based on IVD-PCF and inter-hetSNP logR-based LOH detection (IVD: Inter-Variant Distance, het: heterozygote)
This method reconstructs the normal-pair counts by using the allele counts of the Germline as template
It fills the detected LOH regions with evenly-distributed hetSNPs with the density estimated based on each chromosome in each germline sample
It essentially informs Battenberg of the location of hetSNPs across the genome in the germline sample"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Reconstruct normal-pair allele count files for Germlines</h1>
    <small class="dont-index">Source: <a href="https://github.com/Wedge-Oxford/battenberg/blob/HEAD/R/prepare_wgs_germline.R" class="external-link"><code>R/prepare_wgs_germline.R</code></a></small>
    <div class="hidden name"><code>germline_reconstruct_normal.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Function to generate normal-pair allele count files based on IVD-PCF and inter-hetSNP logR-based LOH detection (IVD: Inter-Variant Distance, het: heterozygote)
This method reconstructs the normal-pair counts by using the allele counts of the Germline as template
It fills the detected LOH regions with evenly-distributed hetSNPs with the density estimated based on each chromosome in each germline sample
It essentially informs Battenberg of the location of hetSNPs across the genome in the germline sample</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">germline_reconstruct_normal</span><span class="op">(</span></span>
<span>  <span class="va">GERMLINENAME</span>,</span>
<span>  <span class="va">NORMALNAME</span>,</span>
<span>  <span class="va">chrom_coord</span>,</span>
<span>  <span class="va">chrom</span>,</span>
<span>  <span class="va">GL_OHET</span>,</span>
<span>  <span class="va">GL_AL</span>,</span>
<span>  <span class="va">GL_AC</span>,</span>
<span>  <span class="va">GL_LogR</span>,</span>
<span>  <span class="va">GAMMA_IVD</span>,</span>
<span>  <span class="va">KMIN_IVD</span>,</span>
<span>  <span class="va">CENTROMERE_NOISE_SEG_SIZE</span>,</span>
<span>  <span class="va">CENTROMERE_DIST</span>,</span>
<span>  <span class="va">MIN_HET_DIST</span>,</span>
<span>  <span class="va">GAMMA_LOGR</span>,</span>
<span>  <span class="va">LENGTH_ADJACENT</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-germlinename">GERMLINENAME<a class="anchor" aria-label="anchor" href="#arg-germlinename"></a></dt>
<dd><p>The germline name used for Battenberg (i.e. the germline BAM file name without the '.bam' extension)</p></dd>


<dt id="arg-normalname">NORMALNAME<a class="anchor" aria-label="anchor" href="#arg-normalname"></a></dt>
<dd><p>The normal name used for naming the generated normal-pair allele counts files</p></dd>


<dt id="arg-chrom-coord">chrom_coord<a class="anchor" aria-label="anchor" href="#arg-chrom-coord"></a></dt>
<dd><p>Full path to the file with chromosome coordinates including start, end and left/right centromere positions</p></dd>


<dt id="arg-chrom">chrom<a class="anchor" aria-label="anchor" href="#arg-chrom"></a></dt>
<dd><p>Chromosome number for which normal-pair will be reconstructed</p></dd>


<dt id="arg-gl-ohet">GL_OHET<a class="anchor" aria-label="anchor" href="#arg-gl-ohet"></a></dt>
<dd><p>List of observed heterozygous SNPs across all chromosomes generated within the germline_baf_logR function</p></dd>


<dt id="arg-gl-al">GL_AL<a class="anchor" aria-label="anchor" href="#arg-gl-al"></a></dt>
<dd><p>List of alleles at SNPs across all chromosomes generated within the germline_baf_logR function</p></dd>


<dt id="arg-gl-ac">GL_AC<a class="anchor" aria-label="anchor" href="#arg-gl-ac"></a></dt>
<dd><p>List of allele counts at SNPs across all chromosomes generated within the germline_baf_logR function</p></dd>


<dt id="arg-gl-logr">GL_LogR<a class="anchor" aria-label="anchor" href="#arg-gl-logr"></a></dt>
<dd><p>Dataframe of genomewide LogR values for SNPs across all chromosomes generated within the germline_baf_logR function</p></dd>


<dt id="arg-gamma-ivd">GAMMA_IVD<a class="anchor" aria-label="anchor" href="#arg-gamma-ivd"></a></dt>
<dd><p>The PCF gamma value for segmentation of 1000G hetSNP IVD values (Default 1e5)</p></dd>


<dt id="arg-kmin-ivd">KMIN_IVD<a class="anchor" aria-label="anchor" href="#arg-kmin-ivd"></a></dt>
<dd><p>The min number of SNPs to support a segment in PCF of 1000G hetSNP IVD values (Default 50)</p></dd>


<dt id="arg-centromere-dist">CENTROMERE_DIST<a class="anchor" aria-label="anchor" href="#arg-centromere-dist"></a></dt>
<dd><p>The minimum distance from the centromere to ignore in analysis due to the noisy nature of data in the vicinity of centromeres (Default 5e5)</p></dd>


<dt id="arg-min-het-dist">MIN_HET_DIST<a class="anchor" aria-label="anchor" href="#arg-min-het-dist"></a></dt>
<dd><p>The minimum distance for detecting higher resolution inter-hetSNP regions with potential LOH while accounting for inherent homozygote stretches (Default 1e5)</p></dd>


<dt id="arg-gamma-logr">GAMMA_LOGR<a class="anchor" aria-label="anchor" href="#arg-gamma-logr"></a></dt>
<dd><p>The PCF gamma value for confirming LOH within each inter-hetSNP candidate segment (Default 100)</p></dd>


<dt id="arg-length-adjacent">LENGTH_ADJACENT<a class="anchor" aria-label="anchor" href="#arg-length-adjacent"></a></dt>
<dd><p>The length of adjacent regions either side of a candidate inter-hetSNP LOH region to be plotted (Default 5e4)</p></dd>

</dl></div>
    <div id="author">
    <h2>Author</h2>
    <p>Naser Ansari-Pour (BDI, Oxford)</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

