<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Package index â€¢ Battenberg</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="GetChromosomeBAFs.html">GetChromosomeBAFs()</a></code> </p>
        </td>
        <td><p>Morphs phased SNPs from WGS input into haplotype blocks</p></td>
      </tr><tr><td>
          <p><code><a href="GetChromosomeBAFs_SNP6.html">GetChromosomeBAFs_SNP6()</a></code> </p>
        </td>
        <td><p>Morphs phased SNPs from SNP6 input into haplotype blocks</p></td>
      </tr><tr><td>
          <p><code><a href="allele_ratio_plot.html">allele_ratio_plot()</a></code> </p>
        </td>
        <td><p>Plot allele ratios from raw segmented data</p></td>
      </tr><tr><td>
          <p><code><a href="battenberg.html">battenberg()</a></code> </p>
        </td>
        <td><p>Run the Battenberg pipeline</p></td>
      </tr><tr><td>
          <p><code><a href="calc_psi_t.html">calc_psi_t()</a></code> </p>
        </td>
        <td><p>Calculate psi based on a reference segment and its associated logr</p></td>
      </tr><tr><td>
          <p><code><a href="calc_rho_psi_refit.html">calc_rho_psi_refit()</a></code> </p>
        </td>
        <td><p>Calculate rho and psi values from a refit suggestion</p></td>
      </tr><tr><td>
          <p><code><a href="calculate_solution_fast.html">calculate_solution_fast()</a></code> </p>
        </td>
        <td><p>Fast solution calculation (vectorized and optimized)</p></td>
      </tr><tr><td>
          <p><code><a href="callChrXsubclones.html">callChrXsubclones()</a></code> </p>
        </td>
        <td><p>Fit ChrX subclonal copy number (male only)</p></td>
      </tr><tr><td>
          <p><code><a href="callSubclones.html">callSubclones()</a></code> </p>
        </td>
        <td><p>Fit subclonal copy number</p></td>
      </tr><tr><td>
          <p><code><a href="call_multisample_MSAI.html">call_multisample_MSAI()</a></code> </p>
        </td>
        <td><p>Generates haplotype blocks, MSAI results, and plots from phasing information contained in multisample Battenberg runs</p></td>
      </tr><tr><td>
          <p><code><a href="cel2baf.logr.html">cel2baf.logr()</a></code> </p>
        </td>
        <td><p>Transform cel files into BAF and LogR</p></td>
      </tr><tr><td>
          <p><code><a href="cell_line_baf_logR.html">cell_line_baf_logR()</a></code> </p>
        </td>
        <td><p>Obtain BAF and LogR from the Cell line (tumour only) allele counts</p></td>
      </tr><tr><td>
          <p><code><a href="cell_line_reconstruct_normal.html">cell_line_reconstruct_normal()</a></code> </p>
        </td>
        <td><p>Reconstruct normal-pair allele count files for cell lines</p></td>
      </tr><tr><td>
          <p><code><a href="check.imputeinfofile.html">check.imputeinfofile()</a></code> </p>
        </td>
        <td><p>Check impute info file consistency</p></td>
      </tr><tr><td>
          <p><code><a href="cnfit_to_refit_suggestions.html">cnfit_to_refit_suggestions()</a></code> </p>
        </td>
        <td><p>Create refit suggestions for a fit copy number profile</p></td>
      </tr><tr><td>
          <p><code><a href="combine.baf.files.html">combine.baf.files()</a></code> </p>
        </td>
        <td><p>Combines all separate BAF files per chromosome into a single file</p></td>
      </tr><tr><td>
          <p><code><a href="combine.impute.output.html">combine.impute.output()</a></code> </p>
        </td>
        <td><p>Concatenate the impute output generated for each of the regions.</p></td>
      </tr><tr><td>
          <p><code><a href="convert.impute.input.to.beagle.input.html">convert.impute.input.to.beagle.input()</a></code> </p>
        </td>
        <td><p>Converts impute input to a beagle input</p></td>
      </tr><tr><td>
          <p><code><a href="coverage_plot.html">coverage_plot()</a></code> </p>
        </td>
        <td><p>Plot relative coverage of tumour and normal</p></td>
      </tr><tr><td>
          <p><code><a href="create_smart_search_order.html">create_smart_search_order()</a></code> </p>
        </td>
        <td><p>Create smart search order - best regions first</p></td>
      </tr><tr><td>
          <p><code><a href="find_centroid_of_global_minima.html">find_centroid_of_global_minima()</a></code> </p>
        </td>
        <td><p>This function is an alternative procedure for finding the optimum (psi, rho) pair. This function first finds all the find all the global optima, and then finds the centroid of this set of globla optima. Then we find the global optimum which is nearest to the centroid. (When the set of global optima is convex, we expect the selected optimum to be at the centroid.)</p></td>
      </tr><tr><td>
          <p><code><a href="fit.copy.number.html">fit.copy.number()</a></code> </p>
        </td>
        <td><p>Fit copy number</p></td>
      </tr><tr><td>
          <p><code><a href="gc.correct.html">gc.correct()</a></code> </p>
        </td>
        <td><p>Correct the LogR estimates for GC content</p></td>
      </tr><tr><td>
          <p><code><a href="gc.correct.wgs.html">gc.correct.wgs()</a></code> </p>
        </td>
        <td><p>Function to correct LogR for waivyness that correlates with GC content</p></td>
      </tr><tr><td>
          <p><code><a href="gc.correct.wgs.germline.html">gc.correct.wgs.germline()</a></code> </p>
        </td>
        <td><p>Function to correct LogR for waivyness that correlates with GC content</p></td>
      </tr><tr><td>
          <p><code><a href="generate.impute.input.snp6.html">generate.impute.input.snp6()</a></code> </p>
        </td>
        <td><p>Prepares data for impute</p></td>
      </tr><tr><td>
          <p><code><a href="generate.impute.input.wgs.html">generate.impute.input.wgs()</a></code> </p>
        </td>
        <td><p>Prepare data for impute</p></td>
      </tr><tr><td>
          <p><code><a href="generate.impute.input.wgs.germline.html">generate.impute.input.wgs.germline()</a></code> </p>
        </td>
        <td><p>Prepare data for impute</p></td>
      </tr><tr><td>
          <p><code><a href="generate_plots_battenberg.html">generate_plots_battenberg()</a></code> </p>
        </td>
        <td><p>Generate plots</p></td>
      </tr><tr><td>
          <p><code><a href="germline_baf_logR.html">germline_baf_logR()</a></code> </p>
        </td>
        <td><p>Obtain BAF and LogR from the Germline allele counts</p></td>
      </tr><tr><td>
          <p><code><a href="germline_reconstruct_normal.html">germline_reconstruct_normal()</a></code> </p>
        </td>
        <td><p>Reconstruct normal-pair allele count files for Germlines</p></td>
      </tr><tr><td>
          <p><code><a href="get.chrom.names.html">get.chrom.names()</a></code> </p>
        </td>
        <td><p>Returns the chromosome names that are supported</p></td>
      </tr><tr><td>
          <p><code><a href="getAlleleCounts.html">getAlleleCounts()</a></code> </p>
        </td>
        <td><p>Obtain allele counts for 1000 Genomes loci through external program alleleCount</p></td>
      </tr><tr><td>
          <p><code><a href="getBAFsAndLogRs.html">getBAFsAndLogRs()</a></code> </p>
        </td>
        <td><p>Obtain BAF and LogR from the allele counts</p></td>
      </tr><tr><td>
          <p><code><a href="get_multisample_phasing.html">get_multisample_phasing()</a></code> </p>
        </td>
        <td><p>Generates phased haplotypes from multisample Battenberg runs</p></td>
      </tr><tr><td>
          <p><code><a href="infer_gender_birdseed.html">infer_gender_birdseed()</a></code> </p>
        </td>
        <td><p>Infer the gender using the birdseed report file</p></td>
      </tr><tr><td>
          <p><code><a href="input_known_haplotypes.html">input_known_haplotypes()</a></code> </p>
        </td>
        <td><p>Combine imputation results with external haplotype blocks</p></td>
      </tr><tr><td>
          <p><code><a href="is_local_minimum_fast.html">is_local_minimum_fast()</a></code> </p>
        </td>
        <td><p>Fast local minimum check (optimized version of original 7x7)</p></td>
      </tr><tr><td>
          <p><code><a href="make_posthoc_plots.html">make_posthoc_plots()</a></code> </p>
        </td>
        <td><p>Function to make additional figures</p></td>
      </tr><tr><td>
          <p><code><a href="mask_high_cn_segments.html">mask_high_cn_segments()</a></code> </p>
        </td>
        <td><p>Mask segments that have a too high CN state</p></td>
      </tr><tr><td>
          <p><code><a href="parse.imputeinfofile.html">parse.imputeinfofile()</a></code> </p>
        </td>
        <td><p>Read in the imputeinfofile.</p></td>
      </tr><tr><td>
          <p><code><a href="plot.haplotype.data.html">plot(<i>&lt;haplotype.data&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot haplotyped SNPs</p></td>
      </tr><tr><td>
          <p><code><a href="prepare_snp6.html">prepare_snp6()</a></code> </p>
        </td>
        <td><p>Prepare SNP6 data for haplotype construction</p></td>
      </tr><tr><td>
          <p><code><a href="prepare_wgs.html">prepare_wgs()</a></code> </p>
        </td>
        <td><p>Prepare WGS data for haplotype construction</p></td>
      </tr><tr><td>
          <p><code><a href="prepare_wgs_cell_line.html">prepare_wgs_cell_line()</a></code> </p>
        </td>
        <td><p>Prepare WGS data of cell line for haplotype construction</p></td>
      </tr><tr><td>
          <p><code><a href="prepare_wgs_germline.html">prepare_wgs_germline()</a></code> </p>
        </td>
        <td><p>Prepare WGS data of germline for haplotype construction</p></td>
      </tr><tr><td>
          <p><code><a href="read_alleleFrequencies.html">read_alleleFrequencies()</a></code> </p>
        </td>
        <td><p>Parser for allele frequencies data</p></td>
      </tr><tr><td>
          <p><code><a href="read_baf.html">read_baf()</a></code> </p>
        </td>
        <td><p>Parser for BAF data</p></td>
      </tr><tr><td>
          <p><code><a href="read_bafsegmented.html">read_bafsegmented()</a></code> </p>
        </td>
        <td><p>Parser for BAFsegmented data</p></td>
      </tr><tr><td>
          <p><code><a href="read_beagle_output.html">read_beagle_output()</a></code> </p>
        </td>
        <td><p>Parser for beagle5 output data</p></td>
      </tr><tr><td>
          <p><code><a href="read_gccontent.html">read_gccontent()</a></code> </p>
        </td>
        <td><p>Parser for GC content reference data</p></td>
      </tr><tr><td>
          <p><code><a href="read_impute_input.html">read_impute_input()</a></code> </p>
        </td>
        <td><p>Parser for impute input data</p></td>
      </tr><tr><td>
          <p><code><a href="read_imputed_output.html">read_imputed_output()</a></code> </p>
        </td>
        <td><p>Parser for imputed genotype data</p></td>
      </tr><tr><td>
          <p><code><a href="read_logr.html">read_logr()</a></code> </p>
        </td>
        <td><p>Parser for logR data</p></td>
      </tr><tr><td>
          <p><code><a href="read_replication.html">read_replication()</a></code> </p>
        </td>
        <td><p>Parser for replication timing reference data</p></td>
      </tr><tr><td>
          <p><code><a href="read_table_generic.html">read_table_generic()</a></code> </p>
        </td>
        <td><p>Generic reading function using the readr R package, tailored for reading in genomic data</p></td>
      </tr><tr><td>
          <p><code><a href="run.beagle5.html">run.beagle5()</a></code> </p>
        </td>
        <td><p>Command to run beagle5</p></td>
      </tr><tr><td>
          <p><code><a href="run.impute.html">run.impute()</a></code> </p>
        </td>
        <td><p>Run impute on the specified inputfile</p></td>
      </tr><tr><td>
          <p><code><a href="runASCAT.html">runASCAT()</a></code> </p>
        </td>
        <td><p>A modified ASCAT main function to fit Battenberg</p></td>
      </tr><tr><td>
          <p><code><a href="runASCAT_enhanced.html">runASCAT_enhanced()</a></code> </p>
        </td>
        <td><p>Key optimizations: 1. Early termination after first good solution (like original) 2. Vectorized distance calculations 3. Optimized constraint checking 4. Smart search ordering (best regions first) 5. Reduced memory allocations</p></td>
      </tr><tr><td>
          <p><code><a href="run_clonal_ASCAT.html">run_clonal_ASCAT()</a></code> </p>
        </td>
        <td><p>ASCAT like function to obtain a clonal copy number profile</p></td>
      </tr><tr><td>
          <p><code><a href="run_haplotyping.html">run_haplotyping()</a></code> </p>
        </td>
        <td><p>Construct haplotypes for a chromosome</p></td>
      </tr><tr><td>
          <p><code><a href="run_haplotyping_germline.html">run_haplotyping_germline()</a></code> </p>
        </td>
        <td><p>Construct haplotypes for a chromosome - germline WGS version</p></td>
      </tr><tr><td>
          <p><code><a href="segment.baf.phased.html">segment.baf.phased()</a></code> </p>
        </td>
        <td><p>Segment BAF, with the possible inclusion of structural variant breakpoints</p></td>
      </tr><tr><td>
          <p><code><a href="segment.baf.phased.legacy.html">segment.baf.phased.legacy()</a></code> </p>
        </td>
        <td><p>Segment the haplotyped and phased data using fastPCF. This is the legacy segmentation function as it was used in the original Battenberg versions</p></td>
      </tr><tr><td>
          <p><code><a href="segment.baf.phased.multisample.html">segment.baf.phased.multisample()</a></code> </p>
        </td>
        <td><p>Segment BAF, with the possible inclusion of structural variant breakpoints</p></td>
      </tr><tr><td>
          <p><code><a href="segment.baf.phased.sv.html">segment.baf.phased.sv()</a></code> </p>
        </td>
        <td><p>Segment BAF with the inclusion of structural variant breakpoints - This function is now deprecated, call segment.baf.phased instead</p></td>
      </tr><tr><td>
          <p><code><a href="split_input_haplotypes.html">split_input_haplotypes()</a></code> </p>
        </td>
        <td><p>Split a single vcf into separate vcfs for each chromosome</p></td>
      </tr><tr><td>
          <p><code><a href="squaresplot.html">squaresplot()</a></code> </p>
        </td>
        <td><p>Plot Battenberg copy number solutions for a segment</p></td>
      </tr><tr><td>
          <p><code><a href="standardiseChrNotation.html">standardiseChrNotation()</a></code> </p>
        </td>
        <td><p>Chromosome notation standardisation (removing 'chr' string from chromosome names - mainly an issue in hg38 BAMs)</p></td>
      </tr><tr><td>
          <p><code><a href="standardiseChrNotation_germline.html">standardiseChrNotation_germline()</a></code> </p>
        </td>
        <td><p>Chromosome notation standardisation (removing 'chr' string from chromosome names - mainly an issue in hg38 BAMs)</p></td>
      </tr><tr><td>
          <p><code><a href="suggest_refit.html">suggest_refit()</a></code> </p>
        </td>
        <td><p>Calculate refit values from a refit suggestion</p></td>
      </tr><tr><td>
          <p><code><a href="totalcn_chrom_plot.html">totalcn_chrom_plot()</a></code> </p>
        </td>
        <td><p>Plot total copy number split per chromosome</p></td>
      </tr><tr><td>
          <p><code><a href="write_battenberg_phasing.html">write_battenberg_phasing()</a></code> </p>
        </td>
        <td><p>Writes the imputation and copy number phased haplotypes to a vcf</p></td>
      </tr><tr><td>
          <p><code><a href="writebeagle.as.impute.html">writebeagle.as.impute()</a></code> </p>
        </td>
        <td><p>Writes output of beagle as output from impute (interface bealge/impute for Battenberg)</p></td>
      </tr><tr><td>
          <p><code><a href="writevcf.beagle.html">writevcf.beagle()</a></code> </p>
        </td>
        <td><p>Writes input file for beagle5</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

