<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prepare WGS data of cell line for haplotype construction — prepare_wgs_cell_line • Battenberg</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Prepare WGS data of cell line for haplotype construction — prepare_wgs_cell_line"><meta property="og:description" content="This function performs part of the Battenberg WGS pipeline: Counting alleles, generating BAF and logR,
reconstructing normal-pair allele counts for the cell line and performing GC content correction."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Prepare WGS data of cell line for haplotype construction</h1>
    <small class="dont-index">Source: <a href="https://github.com/Wedge-Oxford/battenberg/blob/HEAD/R/prepare_wgs_cell_line.R" class="external-link"><code>R/prepare_wgs_cell_line.R</code></a></small>
    <div class="hidden name"><code>prepare_wgs_cell_line.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function performs part of the Battenberg WGS pipeline: Counting alleles, generating BAF and logR,
reconstructing normal-pair allele counts for the cell line and performing GC content correction.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">prepare_wgs_cell_line</span><span class="op">(</span></span>
<span>  <span class="va">chrom_names</span>,</span>
<span>  <span class="va">chrom_coord</span>,</span>
<span>  <span class="va">tumourbam</span>,</span>
<span>  <span class="va">tumourname</span>,</span>
<span>  <span class="va">g1000lociprefix</span>,</span>
<span>  <span class="va">g1000allelesprefix</span>,</span>
<span>  gamma_ivd <span class="op">=</span> <span class="fl">100000</span>,</span>
<span>  kmin_ivd <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  centromere_noise_seg_size <span class="op">=</span> <span class="fl">1000000</span>,</span>
<span>  centromere_dist <span class="op">=</span> <span class="fl">500000</span>,</span>
<span>  min_het_dist <span class="op">=</span> <span class="fl">100000</span>,</span>
<span>  gamma_logr <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  length_adjacent <span class="op">=</span> <span class="fl">50000</span>,</span>
<span>  <span class="va">gccorrectprefix</span>,</span>
<span>  <span class="va">repliccorrectprefix</span>,</span>
<span>  <span class="va">min_base_qual</span>,</span>
<span>  <span class="va">min_map_qual</span>,</span>
<span>  <span class="va">allelecounter_exe</span>,</span>
<span>  <span class="va">min_normal_depth</span>,</span>
<span>  <span class="va">skip_allele_counting</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-chrom-names">chrom_names<a class="anchor" aria-label="anchor" href="#arg-chrom-names"></a></dt>
<dd><p>A vector containing the names of chromosomes to be included</p></dd>


<dt id="arg-tumourbam">tumourbam<a class="anchor" aria-label="anchor" href="#arg-tumourbam"></a></dt>
<dd><p>Full path to the tumour BAM file</p></dd>


<dt id="arg-tumourname">tumourname<a class="anchor" aria-label="anchor" href="#arg-tumourname"></a></dt>
<dd><p>Identifier to be used for tumour output files (i.e. the cell line BAM file name without the '.bam' extension).</p></dd>


<dt id="arg-g-lociprefix">g1000lociprefix<a class="anchor" aria-label="anchor" href="#arg-g-lociprefix"></a></dt>
<dd><p>Prefix path to the 1000 Genomes loci reference files</p></dd>


<dt id="arg-g-allelesprefix">g1000allelesprefix<a class="anchor" aria-label="anchor" href="#arg-g-allelesprefix"></a></dt>
<dd><p>Prefix path to the 1000 Genomes SNP allele reference files</p></dd>


<dt id="arg-gamma-ivd">gamma_ivd<a class="anchor" aria-label="anchor" href="#arg-gamma-ivd"></a></dt>
<dd><p>The PCF gamma value for segmentation of 1000G hetSNP IVD values (Default 1e5).</p></dd>


<dt id="arg-kmin-ivd">kmin_ivd<a class="anchor" aria-label="anchor" href="#arg-kmin-ivd"></a></dt>
<dd><p>The min number of SNPs to support a segment in PCF of 1000G hetSNP IVD values (Default 50)</p></dd>


<dt id="arg-centromere-noise-seg-size">centromere_noise_seg_size<a class="anchor" aria-label="anchor" href="#arg-centromere-noise-seg-size"></a></dt>
<dd><p>The maximum size of PCF segment to be removed as noise when it overlaps with the centromere due to the noisy nature of data (Default 1e6)</p></dd>


<dt id="arg-centromere-dist">centromere_dist<a class="anchor" aria-label="anchor" href="#arg-centromere-dist"></a></dt>
<dd><p>The minimum distance from the centromere to ignore in analysis due to the noisy nature of data in the vicinity of centromeres (Default 5e5)</p></dd>


<dt id="arg-min-het-dist">min_het_dist<a class="anchor" aria-label="anchor" href="#arg-min-het-dist"></a></dt>
<dd><p>The minimum distance for detecting higher resolution inter-hetSNP regions with potential LOH while accounting for inherent homozygote stretches (Default 1e5)</p></dd>


<dt id="arg-gamma-logr">gamma_logr<a class="anchor" aria-label="anchor" href="#arg-gamma-logr"></a></dt>
<dd><p>The PCF gamma value for confirming LOH within each inter-hetSNP candidate segment (Default 100)</p></dd>


<dt id="arg-length-adjacent">length_adjacent<a class="anchor" aria-label="anchor" href="#arg-length-adjacent"></a></dt>
<dd><p>The length of adjacent regions either side of a candidate inter-hetSNP LOH region to be plotted (Default 5e4)</p></dd>


<dt id="arg-gccorrectprefix">gccorrectprefix<a class="anchor" aria-label="anchor" href="#arg-gccorrectprefix"></a></dt>
<dd><p>Prefix path to GC content reference data</p></dd>


<dt id="arg-repliccorrectprefix">repliccorrectprefix<a class="anchor" aria-label="anchor" href="#arg-repliccorrectprefix"></a></dt>
<dd><p>Prefix path to replication timing reference data (supply NULL if no replication timing correction is to be applied)</p></dd>


<dt id="arg-min-base-qual">min_base_qual<a class="anchor" aria-label="anchor" href="#arg-min-base-qual"></a></dt>
<dd><p>Minimum base quality required for a read to be counted</p></dd>


<dt id="arg-min-map-qual">min_map_qual<a class="anchor" aria-label="anchor" href="#arg-min-map-qual"></a></dt>
<dd><p>Minimum mapping quality required for a read to be counted</p></dd>


<dt id="arg-allelecounter-exe">allelecounter_exe<a class="anchor" aria-label="anchor" href="#arg-allelecounter-exe"></a></dt>
<dd><p>Path to the allele counter executable (can be found in $PATH)</p></dd>


<dt id="arg-min-normal-depth">min_normal_depth<a class="anchor" aria-label="anchor" href="#arg-min-normal-depth"></a></dt>
<dd><p>Minimum depth required in the normal for a SNP to be included</p></dd>


<dt id="arg-skip-allele-counting">skip_allele_counting<a class="anchor" aria-label="anchor" href="#arg-skip-allele-counting"></a></dt>
<dd><p>Flag, set to TRUE if allele counting is already complete (files are expected in the working directory on disk)</p></dd>

</dl></div>
    <div id="author">
    <h2>Author</h2>
    <p>Naser Ansari-Pour (BDI, Oxford)</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

