<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>A modified ASCAT main function to fit Battenberg — runASCAT • Battenberg</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="A modified ASCAT main function to fit Battenberg — runASCAT"><meta property="og:description" content="This function returns an initial rho and psi estimate for a clonal copy number fit. It uses an internal distance metric to create a distance matrix.
Using that matrix it will search for a rho and psi combination that yields the least heavy penalty."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Battenberg</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.9.9.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/advanced-usage.html">Advanced Usage and Parameter Optimization</a>
    </li>
    <li>
      <a href="../articles/data-interpretation.html">Data Interpretation and Analysis</a>
    </li>
    <li>
      <a href="../articles/getting-started.html">Getting Started with Battenberg</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/Wedge-Oxford/battenberg/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>A modified ASCAT main function to fit Battenberg</h1>
    <small class="dont-index">Source: <a href="https://github.com/Wedge-Oxford/battenberg/blob/HEAD/R/clonal_ascat.R" class="external-link"><code>R/clonal_ascat.R</code></a></small>
    <div class="hidden name"><code>runASCAT.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function returns an initial rho and psi estimate for a clonal copy number fit. It uses an internal distance metric to create a distance matrix.
Using that matrix it will search for a rho and psi combination that yields the least heavy penalty.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">runASCAT</span><span class="op">(</span></span>
<span>  <span class="va">lrr</span>,</span>
<span>  <span class="va">baf</span>,</span>
<span>  <span class="va">lrrsegmented</span>,</span>
<span>  <span class="va">bafsegmented</span>,</span>
<span>  <span class="va">chromosomes</span>,</span>
<span>  <span class="va">dist_choice</span>,</span>
<span>  distancepng <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  copynumberprofilespng <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  nonroundedprofilepng <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  cnaStatusFile <span class="op">=</span> <span class="st">"copynumber_solution_status.txt"</span>,</span>
<span>  gamma <span class="op">=</span> <span class="fl">0.55</span>,</span>
<span>  <span class="va">allow100percent</span>,</span>
<span>  reliabilityFile <span class="op">=</span> <span class="cn">NA</span>,</span>
<span>  min.ploidy <span class="op">=</span> <span class="fl">1.6</span>,</span>
<span>  max.ploidy <span class="op">=</span> <span class="fl">4.8</span>,</span>
<span>  min.rho <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  max.rho <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  min.goodness <span class="op">=</span> <span class="fl">63</span>,</span>
<span>  uninformative_BAF_threshold <span class="op">=</span> <span class="fl">0.51</span>,</span>
<span>  <span class="va">chr.names</span>,</span>
<span>  analysis <span class="op">=</span> <span class="st">"paired"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-lrr">lrr<a class="anchor" aria-label="anchor" href="#arg-lrr"></a></dt>
<dd><p>(unsegmented) log R, in genomic sequence (all probes), with probe IDs</p></dd>


<dt id="arg-baf">baf<a class="anchor" aria-label="anchor" href="#arg-baf"></a></dt>
<dd><p>(unsegmented) B Allele Frequency, in genomic sequence (all probes), with probe IDs</p></dd>


<dt id="arg-lrrsegmented">lrrsegmented<a class="anchor" aria-label="anchor" href="#arg-lrrsegmented"></a></dt>
<dd><p>log R, segmented, in genomic sequence (all probes), with probe IDs</p></dd>


<dt id="arg-bafsegmented">bafsegmented<a class="anchor" aria-label="anchor" href="#arg-bafsegmented"></a></dt>
<dd><p>B Allele Frequency, segmented, in genomic sequence (only probes heterozygous in germline), with probe IDs</p></dd>


<dt id="arg-chromosomes">chromosomes<a class="anchor" aria-label="anchor" href="#arg-chromosomes"></a></dt>
<dd><p>a list containing c vectors, where c is the number of chromosomes and every vector contains all probe numbers per chromosome</p></dd>


<dt id="arg-dist-choice">dist_choice<a class="anchor" aria-label="anchor" href="#arg-dist-choice"></a></dt>
<dd><p>The distance metric to be used internally to penalise a copy number solution</p></dd>


<dt id="arg-distancepng">distancepng<a class="anchor" aria-label="anchor" href="#arg-distancepng"></a></dt>
<dd><p>if NA: distance is plotted, if filename is given, the plot is written to a .png file (Default NA)</p></dd>


<dt id="arg-copynumberprofilespng">copynumberprofilespng<a class="anchor" aria-label="anchor" href="#arg-copynumberprofilespng"></a></dt>
<dd><p>if NA: possible copy number profiles are plotted, if filename is given, the plot is written to a .png file (Default NA)</p></dd>


<dt id="arg-nonroundedprofilepng">nonroundedprofilepng<a class="anchor" aria-label="anchor" href="#arg-nonroundedprofilepng"></a></dt>
<dd><p>if NA: copy number profile before rounding is plotted (total copy number as well as the copy number of the minor allele), if filename is given, the plot is written to a .png file (Default NA)</p></dd>


<dt id="arg-cnastatusfile">cnaStatusFile<a class="anchor" aria-label="anchor" href="#arg-cnastatusfile"></a></dt>
<dd><p>File where the copy number profile status is written to. This contains either the message "No suitable copy number solution found" or "X copy number solutions found" (Default copynumber_solution_status.txt)</p></dd>


<dt id="arg-gamma">gamma<a class="anchor" aria-label="anchor" href="#arg-gamma"></a></dt>
<dd><p>technology parameter, compaction of Log R profiles (expected decrease in case of deletion in diploid sample, 100 "%" aberrant cells; 1 in ideal case, 0.55 of Illumina 109K arrays) (Default 0.55)</p></dd>


<dt id="arg-allow-percent">allow100percent<a class="anchor" aria-label="anchor" href="#arg-allow-percent"></a></dt>
<dd><p>A boolean whether to allow a 100"%" cellularity solution</p></dd>


<dt id="arg-reliabilityfile">reliabilityFile<a class="anchor" aria-label="anchor" href="#arg-reliabilityfile"></a></dt>
<dd><p>String to where fit reliabilty information should be written. This file contains backtransformed BAF and LogR values for segments using the fitted copy number profile (Default NA)</p></dd>


<dt id="arg-min-ploidy">min.ploidy<a class="anchor" aria-label="anchor" href="#arg-min-ploidy"></a></dt>
<dd><p>The minimum ploidy to consider (Default 1.6)</p></dd>


<dt id="arg-max-ploidy">max.ploidy<a class="anchor" aria-label="anchor" href="#arg-max-ploidy"></a></dt>
<dd><p>The maximum ploidy to consider (Default 4.8)</p></dd>


<dt id="arg-min-rho">min.rho<a class="anchor" aria-label="anchor" href="#arg-min-rho"></a></dt>
<dd><p>The minimum cellularity to consider (Default 0.1)</p></dd>


<dt id="arg-max-rho">max.rho<a class="anchor" aria-label="anchor" href="#arg-max-rho"></a></dt>
<dd><p>The maximum cellularity to consider (Default 1.0)</p></dd>


<dt id="arg-min-goodness">min.goodness<a class="anchor" aria-label="anchor" href="#arg-min-goodness"></a></dt>
<dd><p>The minimum goodness of fit for a solution to have to be considered (Default 63)</p></dd>


<dt id="arg-uninformative-baf-threshold">uninformative_BAF_threshold<a class="anchor" aria-label="anchor" href="#arg-uninformative-baf-threshold"></a></dt>
<dd><p>The threshold beyond which BAF becomes uninformative (Default 0.51)</p></dd>


<dt id="arg-chr-names">chr.names<a class="anchor" aria-label="anchor" href="#arg-chr-names"></a></dt>
<dd><p>A vector with chromosome names used for plotting</p></dd>


<dt id="arg-analysis">analysis<a class="anchor" aria-label="anchor" href="#arg-analysis"></a></dt>
<dd><p>A String representing the type of analysis to be run, this determines whether the distance figure is produced (Default paired)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list with fields psi, rho and ploidy</p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Wedge, Peter Van Loo, Naser Ansari-Pour, Stefan Dentro, Maxime Tarabichi, Jonas Demeulemeester.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

      </footer></div>






  </body></html>

